{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "Data:\n",
        "- https://s-baker.net/resource/hoc/\n",
        "\n",
        "WSD:\n",
        "1. Database ideas\n",
        "  - https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-223\n",
        "  - https://lhncbc.nlm.nih.gov/ii/areas/WSD/collaboration.html#WSD_choices\n",
        "2. Methods\n",
        "  - https://academic.oup.com/bioinformatics/article/26/22/2889/228423\n",
        "  - https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3079-8\n",
        "  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5792196/\n",
        "  - https://pubmed.ncbi.nlm.nih.gov/30811548/\n",
        "  - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7787358/\n",
        "  - https://towardsdatascience.com/lstm-by-example-using-tensorflow-feb0c1968537\n",
        "  - \n",
        "\n",
        "To do: \n",
        "1. Prepare the Hallmarks of Cancer db\n",
        "2. Find all unique words in abstracts\n",
        "  - Filter data?\n",
        "  - Tokenization?\n",
        "3. Apply Jimeno-Yepes et al methods for db creation\n",
        "  - Should I build my own db specific to our documents or just use a pre-existing WSD corpus.\n",
        "  1. Search metathesaurus to find words with more than 1 meaning\n",
        "  2. Remove interpretations (CUI) that do not have MeSH indexing\n",
        "  3. Search for citations that use the CUI of the word in the abstract\n",
        "4. Filter data \n",
        "  - Elim under used CUI\n",
        "  - Elim ambiguous uses (citations that use 2 CUI's of one word)\n",
        "5. Use data set to train WSD\n",
        "6. How do we implement this with our data?\n",
        "  - Replace ambiguous words with non-ambiguous words\n",
        "  - Factor into encoding similar to positional encoding methods in transforms\n"
      ],
      "metadata": {
        "id": "8hXK-MtvH4bc"
      }
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "BZIcvxZE0g9C"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# !git clone https://ChristopheUAB@github.com/ChristopheUAB/WSD-Proj\n",
        "\n",
        "# git clone https://username@github.com/username/repo_name"
      ],
      "metadata": {
        "id": "a4ydAcQGNKDV"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "'''\n",
        "Read and export MSD data to tsv\n",
        "'''\n",
        "\n",
        "from google.colab import files\n",
        "import os\n",
        "import re\n",
        "import pandas as pd\n",
        "from io import StringIO\n",
        "\n",
        "class MSD_Load:\n",
        "  def UploadMSDFiles():\n",
        "    uploaded = files.upload()\n",
        "\n",
        "  def FilePaths(ParentPath):\n",
        "    Paths = []\n",
        "    with os.scandir('/content/') as entries:\n",
        "        for entry in entries:\n",
        "          if entry.name.endswith('.arff'):\n",
        "            Paths.append(ParentPath + entry.name)\n",
        "    return Paths\n",
        "  \n",
        "  def ExtractDat(Paths):\n",
        "    FinalData = ''\n",
        "    CUIDic = {}\n",
        "    for entry in Paths:\n",
        "      match = re.search(r'/content/(.*)_pmids_tagged.arff',entry).group(1)\n",
        "      with open(entry, 'r', encoding = \"ISO-8859-1\") as f:\n",
        "        data = f.read()\n",
        "        \n",
        "        # Extract CUI, attributes, and data\n",
        "        CUIData = re.search(r'@RELATION (.*)\\n', data).group(1).split('_')\n",
        "        Attributes = re.findall(r'@ATTRIBUTE (.*)\\n', data)\n",
        "        FullData = re.search(r'@DATA\\n(.*)', data, re.DOTALL).group(1)\n",
        "\n",
        "        # Match CUI and M#s\n",
        "        Classes = [i for i in Attributes if 'class' in i]\n",
        "        Index = Attributes.index(Classes[0])\n",
        "        MVals = re.search(r'class {(.*)}', Classes[0])\n",
        "        MValsList = MVals.group(1).split(', ')\n",
        "\n",
        "        # Replace M#s with correct CUIs in FullData\n",
        "        c = 0\n",
        "        for CUI in CUIData:\n",
        "          FullData = FullData.replace(MValsList[c] + '\\n',CUI + '\\n')\n",
        "          c += 1\n",
        "\n",
        "        # Insert ambiguous at the start of all lines\n",
        "        FullData = FullData.split('\\n')\n",
        "        del FullData[-1]\n",
        "        c = 0\n",
        "        for line in FullData:\n",
        "          FullData[c] = match + ',' + line\n",
        "          c += 1\n",
        "        FullData = '\\n'.join(FullData)\n",
        "\n",
        "        # Append to data list\n",
        "        FinalData += FullData + '\\n'\n",
        "\n",
        "        CUIDic[match] = CUIData\n",
        "\n",
        "        f.close()\n",
        "\n",
        "    # Join data\n",
        "    Attributes[Index] = 'CUI'\n",
        "    AttributesStr = 'ambiguous word,' + \",\".join(Attributes) + '\\n'\n",
        "    FinalData = AttributesStr + FinalData\n",
        "    return FinalData, CUIDic\n",
        "\n",
        "  def String2DF(String):\n",
        "    # Just use splits\n",
        "    String = String.split('\\n')\n",
        "\n",
        "    # Change relevant commas to tabs\n",
        "    FinalString = ''\n",
        "    c = 0\n",
        "    for element in String:\n",
        "      ElementList = element.split('\"')\n",
        "      if len(ElementList)-3 != 0:\n",
        "        FinalString += ElementList[0].replace(',', '\\t') + '\\n'\n",
        "      else:\n",
        "        ElementList[0] = ElementList[0].replace(',', '\\t')\n",
        "        ElementList[2] = ElementList[2].replace(',', '\\t')\n",
        "        FinalString += '\"'.join(ElementList) + '\\n'\n",
        "      c += 1\n",
        "\n",
        "    # Convert tab seperated values to a df\n",
        "    df = pd.read_csv(StringIO(FinalString), sep = '\\t')\n",
        "    return df\n",
        "  \n",
        "  # df.to_csv('MSDFile.tsv',sep='\\t',encoding=\"ISO-8859-1\")\n",
        "  def WriteTSV(df, FileName):\n",
        "    df.to_csv(FileName,sep='\\t',encoding=\"ISO-8859-1\")\n",
        "    return\n",
        "\n",
        "  def ReadTSV(Path):\n",
        "    df = pd.read_csv(Path, sep = '\\t',encoding=\"ISO-8859-1\")\n",
        "    return df\n",
        "\n",
        "MSD_Load.UploadMSDFiles()\n",
        "paths = MSD_Load.FilePaths('/content/')\n",
        "[Data, CUIDic] = MSD_Load.ExtractDat(paths)\n",
        "df = MSD_Load.String2DF(Data)\n",
        "# MSD_Load.WriteTSV(df, 'MSDFile.tsv')\n",
        "# df = MSD_Load.ReadTSV('/content/MSDFile.tsv')"
      ],
      "metadata": {
        "id": "wOzC79vIphgi",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "89a1d223-3094-45d0-a2f6-1d394be284ca"
      },
      "execution_count": 44,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.HTML object>"
            ],
            "text/html": [
              "\n",
              "     <input type=\"file\" id=\"files-7932885a-fbe9-4c35-9b91-394ae47c6a47\" name=\"files[]\" multiple disabled\n",
              "        style=\"border:none\" />\n",
              "     <output id=\"result-7932885a-fbe9-4c35-9b91-394ae47c6a47\">\n",
              "      Upload widget is only available when the cell has been executed in the\n",
              "      current browser session. Please rerun this cell to enable.\n",
              "      </output>\n",
              "      <script>// Copyright 2017 Google LLC\n",
              "//\n",
              "// Licensed under the Apache License, Version 2.0 (the \"License\");\n",
              "// you may not use this file except in compliance with the License.\n",
              "// You may obtain a copy of the License at\n",
              "//\n",
              "//      http://www.apache.org/licenses/LICENSE-2.0\n",
              "//\n",
              "// Unless required by applicable law or agreed to in writing, software\n",
              "// distributed under the License is distributed on an \"AS IS\" BASIS,\n",
              "// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
              "// See the License for the specific language governing permissions and\n",
              "// limitations under the License.\n",
              "\n",
              "/**\n",
              " * @fileoverview Helpers for google.colab Python module.\n",
              " */\n",
              "(function(scope) {\n",
              "function span(text, styleAttributes = {}) {\n",
              "  const element = document.createElement('span');\n",
              "  element.textContent = text;\n",
              "  for (const key of Object.keys(styleAttributes)) {\n",
              "    element.style[key] = styleAttributes[key];\n",
              "  }\n",
              "  return element;\n",
              "}\n",
              "\n",
              "// Max number of bytes which will be uploaded at a time.\n",
              "const MAX_PAYLOAD_SIZE = 100 * 1024;\n",
              "\n",
              "function _uploadFiles(inputId, outputId) {\n",
              "  const steps = uploadFilesStep(inputId, outputId);\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  // Cache steps on the outputElement to make it available for the next call\n",
              "  // to uploadFilesContinue from Python.\n",
              "  outputElement.steps = steps;\n",
              "\n",
              "  return _uploadFilesContinue(outputId);\n",
              "}\n",
              "\n",
              "// This is roughly an async generator (not supported in the browser yet),\n",
              "// where there are multiple asynchronous steps and the Python side is going\n",
              "// to poll for completion of each step.\n",
              "// This uses a Promise to block the python side on completion of each step,\n",
              "// then passes the result of the previous step as the input to the next step.\n",
              "function _uploadFilesContinue(outputId) {\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  const steps = outputElement.steps;\n",
              "\n",
              "  const next = steps.next(outputElement.lastPromiseValue);\n",
              "  return Promise.resolve(next.value.promise).then((value) => {\n",
              "    // Cache the last promise value to make it available to the next\n",
              "    // step of the generator.\n",
              "    outputElement.lastPromiseValue = value;\n",
              "    return next.value.response;\n",
              "  });\n",
              "}\n",
              "\n",
              "/**\n",
              " * Generator function which is called between each async step of the upload\n",
              " * process.\n",
              " * @param {string} inputId Element ID of the input file picker element.\n",
              " * @param {string} outputId Element ID of the output display.\n",
              " * @return {!Iterable<!Object>} Iterable of next steps.\n",
              " */\n",
              "function* uploadFilesStep(inputId, outputId) {\n",
              "  const inputElement = document.getElementById(inputId);\n",
              "  inputElement.disabled = false;\n",
              "\n",
              "  const outputElement = document.getElementById(outputId);\n",
              "  outputElement.innerHTML = '';\n",
              "\n",
              "  const pickedPromise = new Promise((resolve) => {\n",
              "    inputElement.addEventListener('change', (e) => {\n",
              "      resolve(e.target.files);\n",
              "    });\n",
              "  });\n",
              "\n",
              "  const cancel = document.createElement('button');\n",
              "  inputElement.parentElement.appendChild(cancel);\n",
              "  cancel.textContent = 'Cancel upload';\n",
              "  const cancelPromise = new Promise((resolve) => {\n",
              "    cancel.onclick = () => {\n",
              "      resolve(null);\n",
              "    };\n",
              "  });\n",
              "\n",
              "  // Wait for the user to pick the files.\n",
              "  const files = yield {\n",
              "    promise: Promise.race([pickedPromise, cancelPromise]),\n",
              "    response: {\n",
              "      action: 'starting',\n",
              "    }\n",
              "  };\n",
              "\n",
              "  cancel.remove();\n",
              "\n",
              "  // Disable the input element since further picks are not allowed.\n",
              "  inputElement.disabled = true;\n",
              "\n",
              "  if (!files) {\n",
              "    return {\n",
              "      response: {\n",
              "        action: 'complete',\n",
              "      }\n",
              "    };\n",
              "  }\n",
              "\n",
              "  for (const file of files) {\n",
              "    const li = document.createElement('li');\n",
              "    li.append(span(file.name, {fontWeight: 'bold'}));\n",
              "    li.append(span(\n",
              "        `(${file.type || 'n/a'}) - ${file.size} bytes, ` +\n",
              "        `last modified: ${\n",
              "            file.lastModifiedDate ? file.lastModifiedDate.toLocaleDateString() :\n",
              "                                    'n/a'} - `));\n",
              "    const percent = span('0% done');\n",
              "    li.appendChild(percent);\n",
              "\n",
              "    outputElement.appendChild(li);\n",
              "\n",
              "    const fileDataPromise = new Promise((resolve) => {\n",
              "      const reader = new FileReader();\n",
              "      reader.onload = (e) => {\n",
              "        resolve(e.target.result);\n",
              "      };\n",
              "      reader.readAsArrayBuffer(file);\n",
              "    });\n",
              "    // Wait for the data to be ready.\n",
              "    let fileData = yield {\n",
              "      promise: fileDataPromise,\n",
              "      response: {\n",
              "        action: 'continue',\n",
              "      }\n",
              "    };\n",
              "\n",
              "    // Use a chunked sending to avoid message size limits. See b/62115660.\n",
              "    let position = 0;\n",
              "    do {\n",
              "      const length = Math.min(fileData.byteLength - position, MAX_PAYLOAD_SIZE);\n",
              "      const chunk = new Uint8Array(fileData, position, length);\n",
              "      position += length;\n",
              "\n",
              "      const base64 = btoa(String.fromCharCode.apply(null, chunk));\n",
              "      yield {\n",
              "        response: {\n",
              "          action: 'append',\n",
              "          file: file.name,\n",
              "          data: base64,\n",
              "        },\n",
              "      };\n",
              "\n",
              "      let percentDone = fileData.byteLength === 0 ?\n",
              "          100 :\n",
              "          Math.round((position / fileData.byteLength) * 100);\n",
              "      percent.textContent = `${percentDone}% done`;\n",
              "\n",
              "    } while (position < fileData.byteLength);\n",
              "  }\n",
              "\n",
              "  // All done.\n",
              "  yield {\n",
              "    response: {\n",
              "      action: 'complete',\n",
              "    }\n",
              "  };\n",
              "}\n",
              "\n",
              "scope.google = scope.google || {};\n",
              "scope.google.colab = scope.google.colab || {};\n",
              "scope.google.colab._files = {\n",
              "  _uploadFiles,\n",
              "  _uploadFilesContinue,\n",
              "};\n",
              "})(self);\n",
              "</script> "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Saving AA_pmids_tagged.arff to AA_pmids_tagged (1).arff\n",
            "Saving ADA_pmids_tagged.arff to ADA_pmids_tagged (1).arff\n",
            "Saving ADH_pmids_tagged.arff to ADH_pmids_tagged (1).arff\n",
            "Saving ADP_pmids_tagged.arff to ADP_pmids_tagged (1).arff\n",
            "Saving Adrenal_pmids_tagged.arff to Adrenal_pmids_tagged (1).arff\n",
            "Saving Ala_pmids_tagged.arff to Ala_pmids_tagged (1).arff\n",
            "Saving ALS_pmids_tagged.arff to ALS_pmids_tagged (1).arff\n",
            "Saving ANA_pmids_tagged.arff to ANA_pmids_tagged (1).arff\n",
            "Saving Arteriovenous Anastomoses_pmids_tagged.arff to Arteriovenous Anastomoses_pmids_tagged (1).arff\n",
            "Saving Astragalus_pmids_tagged.arff to Astragalus_pmids_tagged (1).arff\n",
            "Saving BAT_pmids_tagged.arff to BAT_pmids_tagged (1).arff\n",
            "Saving B-Cell Leukemia_pmids_tagged.arff to B-Cell Leukemia_pmids_tagged (1).arff\n",
            "Saving benchmark_mesh.txt to benchmark_mesh (1).txt\n",
            "Saving BLM_pmids_tagged.arff to BLM_pmids_tagged (1).arff\n",
            "Saving Borrelia_pmids_tagged.arff to Borrelia_pmids_tagged (1).arff\n",
            "Saving BPD_pmids_tagged.arff to BPD_pmids_tagged (1).arff\n",
            "Saving BR_pmids_tagged.arff to BR_pmids_tagged (1).arff\n",
            "Saving Brucella abortus_pmids_tagged.arff to Brucella abortus_pmids_tagged (1).arff\n",
            "Saving BSA_pmids_tagged.arff to BSA_pmids_tagged (1).arff\n",
            "Saving BSE_pmids_tagged.arff to BSE_pmids_tagged (1).arff\n",
            "Saving Ca_pmids_tagged.arff to Ca_pmids_tagged (1).arff\n",
            "Saving CAD_pmids_tagged.arff to CAD_pmids_tagged (1).arff\n",
            "Saving Callus_pmids_tagged.arff to Callus_pmids_tagged (1).arff\n",
            "Saving CAM_pmids_tagged.arff to CAM_pmids_tagged (1).arff\n",
            "Saving Cardiac pacemaker_pmids_tagged.arff to Cardiac pacemaker_pmids_tagged (1).arff\n",
            "Saving CCD_pmids_tagged.arff to CCD_pmids_tagged (1).arff\n",
            "Saving CCl4_pmids_tagged.arff to CCl4_pmids_tagged (1).arff\n",
            "Saving CDA_pmids_tagged.arff to CDA_pmids_tagged (1).arff\n",
            "Saving CDR_pmids_tagged.arff to CDR_pmids_tagged (1).arff\n",
            "Saving Cell_pmids_tagged.arff to Cell_pmids_tagged (1).arff\n",
            "Saving Cement_pmids_tagged.arff to Cement_pmids_tagged (1).arff\n",
            "Saving CH_pmids_tagged.arff to CH_pmids_tagged (1).arff\n",
            "Saving Cholera_pmids_tagged.arff to Cholera_pmids_tagged (1).arff\n",
            "Saving CI_pmids_tagged.arff to CI_pmids_tagged (1).arff\n",
            "Saving Cilia_pmids_tagged.arff to Cilia_pmids_tagged (1).arff\n",
            "Saving CIS_pmids_tagged.arff to CIS_pmids_tagged (1).arff\n",
            "Saving CLS_pmids_tagged.arff to CLS_pmids_tagged (1).arff\n",
            "Saving CNS_pmids_tagged.arff to CNS_pmids_tagged (1).arff\n",
            "Saving Coffee_pmids_tagged.arff to Coffee_pmids_tagged (1).arff\n",
            "Saving Cold_pmids_tagged.arff to Cold_pmids_tagged (1).arff\n",
            "Saving Compliance_pmids_tagged.arff to Compliance_pmids_tagged (1).arff\n",
            "Saving Cortex_pmids_tagged.arff to Cortex_pmids_tagged (1).arff\n",
            "Saving Cortical_pmids_tagged.arff to Cortical_pmids_tagged (1).arff\n",
            "Saving CP_pmids_tagged.arff to CP_pmids_tagged (1).arff\n",
            "Saving CPDD_pmids_tagged.arff to CPDD_pmids_tagged (1).arff\n",
            "Saving Crack_pmids_tagged.arff to Crack_pmids_tagged (1).arff\n",
            "Saving CRF_pmids_tagged.arff to CRF_pmids_tagged (1).arff\n",
            "Saving cRNA_pmids_tagged.arff to cRNA_pmids_tagged (1).arff\n",
            "Saving Crown_pmids_tagged.arff to Crown_pmids_tagged (1).arff\n",
            "Saving CTX_pmids_tagged.arff to CTX_pmids_tagged (1).arff\n",
            "Saving DAT_pmids_tagged.arff to DAT_pmids_tagged (1).arff\n",
            "Saving DBA_pmids_tagged.arff to DBA_pmids_tagged (1).arff\n",
            "Saving dC_pmids_tagged.arff to dC_pmids_tagged (1).arff\n",
            "Saving DDD_pmids_tagged.arff to DDD_pmids_tagged (1).arff\n",
            "Saving DDS_pmids_tagged.arff to DDS_pmids_tagged (1).arff\n",
            "Saving DE_pmids_tagged.arff to DE_pmids_tagged (1).arff\n",
            "Saving DI_pmids_tagged.arff to DI_pmids_tagged (1).arff\n",
            "Saving Digestive_pmids_tagged.arff to Digestive_pmids_tagged (1).arff\n",
            "Saving DON_pmids_tagged.arff to DON_pmids_tagged (1).arff\n",
            "Saving drinking_pmids_tagged.arff to drinking_pmids_tagged (1).arff\n",
            "Saving eCG_pmids_tagged.arff to eCG_pmids_tagged (1).arff\n",
            "Saving Eels_pmids_tagged.arff to Eels_pmids_tagged (1).arff\n",
            "Saving EGG_pmids_tagged.arff to EGG_pmids_tagged (1).arff\n",
            "Saving EM_pmids_tagged.arff to EM_pmids_tagged (1).arff\n",
            "Saving EMS_pmids_tagged.arff to EMS_pmids_tagged (1).arff\n",
            "Saving Epi_pmids_tagged.arff to Epi_pmids_tagged (1).arff\n",
            "Saving ERP_pmids_tagged.arff to ERP_pmids_tagged (1).arff\n",
            "Saving ERUPTION_pmids_tagged.arff to ERUPTION_pmids_tagged (1).arff\n",
            "Saving Erythrocytes_pmids_tagged.arff to Erythrocytes_pmids_tagged (1).arff\n",
            "Saving Exercises_pmids_tagged.arff to Exercises_pmids_tagged (1).arff\n",
            "Saving FA_pmids_tagged.arff to FA_pmids_tagged (1).arff\n",
            "Saving Familial Adenomatous Polyposis_pmids_tagged.arff to Familial Adenomatous Polyposis_pmids_tagged (1).arff\n",
            "Saving FAS_pmids_tagged.arff to FAS_pmids_tagged (1).arff\n",
            "Saving Fe_pmids_tagged.arff to Fe_pmids_tagged (1).arff\n",
            "Saving Fish_pmids_tagged.arff to Fish_pmids_tagged (1).arff\n",
            "Saving Follicle_pmids_tagged.arff to Follicle_pmids_tagged (1).arff\n",
            "Saving Follicles_pmids_tagged.arff to Follicles_pmids_tagged (1).arff\n",
            "Saving FTC_pmids_tagged.arff to FTC_pmids_tagged (1).arff\n",
            "Saving GAG_pmids_tagged.arff to GAG_pmids_tagged (1).arff\n",
            "Saving Gamma-Interferon_pmids_tagged.arff to Gamma-Interferon_pmids_tagged (1).arff\n",
            "Saving Ganglion_pmids_tagged.arff to Ganglion_pmids_tagged (1).arff\n",
            "Saving Gas_pmids_tagged.arff to Gas_pmids_tagged (1).arff\n",
            "Saving Glycoside_pmids_tagged.arff to Glycoside_pmids_tagged (1).arff\n",
            "Saving Haemophilus ducreyi_pmids_tagged.arff to Haemophilus ducreyi_pmids_tagged (1).arff\n",
            "Saving HCl_pmids_tagged.arff to HCl_pmids_tagged (1).arff\n",
            "Saving Hemlock_pmids_tagged.arff to Hemlock_pmids_tagged (1).arff\n",
            "Saving Heregulin_pmids_tagged.arff to Heregulin_pmids_tagged (1).arff\n",
            "Saving HGF_pmids_tagged.arff to HGF_pmids_tagged (1).arff\n",
            "Saving HHV 8_pmids_tagged.arff to HHV 8_pmids_tagged (1).arff\n",
            "Saving Hip_pmids_tagged.arff to Hip_pmids_tagged (1).arff\n",
            "Saving HIV_pmids_tagged.arff to HIV_pmids_tagged (1).arff\n",
            "Saving HPS_pmids_tagged.arff to HPS_pmids_tagged (1).arff\n",
            "Saving HR_pmids_tagged.arff to HR_pmids_tagged (1).arff\n",
            "Saving Hybridization_pmids_tagged.arff to Hybridization_pmids_tagged (1).arff\n",
            "Saving IA_pmids_tagged.arff to IA_pmids_tagged (1).arff\n",
            "Saving Ice_pmids_tagged.arff to Ice_pmids_tagged (1).arff\n",
            "Saving INDO_pmids_tagged.arff to INDO_pmids_tagged (1).arff\n",
            "Saving Ion_pmids_tagged.arff to Ion_pmids_tagged (1).arff\n",
            "Saving IP_pmids_tagged.arff to IP_pmids_tagged (1).arff\n",
            "Saving Iris_pmids_tagged.arff to Iris_pmids_tagged (1).arff\n",
            "Saving ITP_pmids_tagged.arff to ITP_pmids_tagged (1).arff\n",
            "Saving JP_pmids_tagged.arff to JP_pmids_tagged (1).arff\n",
            "Saving LABOR_pmids_tagged.arff to LABOR_pmids_tagged (1).arff\n",
            "Saving Lactation_pmids_tagged.arff to Lactation_pmids_tagged (1).arff\n",
            "Saving Language_pmids_tagged.arff to Language_pmids_tagged (1).arff\n",
            "Saving Laryngeal_pmids_tagged.arff to Laryngeal_pmids_tagged (1).arff\n",
            "Saving Lawsonia_pmids_tagged.arff to Lawsonia_pmids_tagged (1).arff\n",
            "Saving Leishmaniasis_pmids_tagged.arff to Leishmaniasis_pmids_tagged (1).arff\n",
            "Saving lens_pmids_tagged.arff to lens_pmids_tagged (1).arff\n",
            "Saving Lupus_pmids_tagged.arff to Lupus_pmids_tagged (1).arff\n",
            "Saving lymphogranulomatosis_pmids_tagged.arff to lymphogranulomatosis_pmids_tagged (1).arff\n",
            "Saving MAF_pmids_tagged.arff to MAF_pmids_tagged (1).arff\n",
            "Saving Malaria_pmids_tagged.arff to Malaria_pmids_tagged (1).arff\n",
            "Saving MBP_pmids_tagged.arff to MBP_pmids_tagged (1).arff\n",
            "Saving MCC_pmids_tagged.arff to MCC_pmids_tagged (1).arff\n",
            "Saving Medullary_pmids_tagged.arff to Medullary_pmids_tagged (1).arff\n",
            "Saving MHC_pmids_tagged.arff to MHC_pmids_tagged (1).arff\n",
            "Saving Milk_pmids_tagged.arff to Milk_pmids_tagged (1).arff\n",
            "Saving Moles_pmids_tagged.arff to Moles_pmids_tagged (1).arff\n",
            "Saving MRS_pmids_tagged.arff to MRS_pmids_tagged (1).arff\n",
            "Saving Murine sarcoma virus_pmids_tagged.arff to Murine sarcoma virus_pmids_tagged (1).arff\n",
            "Saving NBS_pmids_tagged.arff to NBS_pmids_tagged (1).arff\n",
            "Saving NEUROFIBROMATOSIS_pmids_tagged.arff to NEUROFIBROMATOSIS_pmids_tagged (1).arff\n",
            "Saving NM_pmids_tagged.arff to NM_pmids_tagged (1).arff\n",
            "Saving NPC_pmids_tagged.arff to NPC_pmids_tagged (1).arff\n",
            "Saving Nurse_pmids_tagged.arff to Nurse_pmids_tagged (1).arff\n",
            "Saving Nursing_pmids_tagged.arff to Nursing_pmids_tagged (1).arff\n",
            "Saving OCD_pmids_tagged.arff to OCD_pmids_tagged (1).arff\n",
            "Saving OH_pmids_tagged.arff to OH_pmids_tagged (1).arff\n",
            "Saving Orf_pmids_tagged.arff to Orf_pmids_tagged (1).arff\n",
            "Saving ORI_pmids_tagged.arff to ORI_pmids_tagged (1).arff\n",
            "Saving PAC_pmids_tagged.arff to PAC_pmids_tagged (1).arff\n",
            "Saving PAF_pmids_tagged.arff to PAF_pmids_tagged (1).arff\n",
            "Saving Parotitis_pmids_tagged.arff to Parotitis_pmids_tagged (1).arff\n",
            "Saving PCA_pmids_tagged.arff to PCA_pmids_tagged (1).arff\n",
            "Saving PCB_pmids_tagged.arff to PCB_pmids_tagged (1).arff\n",
            "Saving PCD_pmids_tagged.arff to PCD_pmids_tagged (1).arff\n",
            "Saving PCP_pmids_tagged.arff to PCP_pmids_tagged (1).arff\n",
            "Saving PEP_pmids_tagged.arff to PEP_pmids_tagged (1).arff\n",
            "Saving PHA_pmids_tagged.arff to PHA_pmids_tagged (1).arff\n",
            "Saving Pharmaceutical_pmids_tagged.arff to Pharmaceutical_pmids_tagged (1).arff\n",
            "Saving Phosphorus_pmids_tagged.arff to Phosphorus_pmids_tagged (1).arff\n",
            "Saving Phosphorylase_pmids_tagged.arff to Phosphorylase_pmids_tagged (1).arff\n",
            "Saving pI_pmids_tagged.arff to pI_pmids_tagged (1).arff\n",
            "Saving Plague_pmids_tagged.arff to Plague_pmids_tagged (1).arff\n",
            "Saving Plaque_pmids_tagged.arff to Plaque_pmids_tagged (1).arff\n",
            "Saving Platelet_pmids_tagged.arff to Platelet_pmids_tagged (1).arff\n",
            "Saving Pleuropneumonia_pmids_tagged.arff to Pleuropneumonia_pmids_tagged (1).arff\n",
            "Saving Pneumocystis_pmids_tagged.arff to Pneumocystis_pmids_tagged (1).arff\n",
            "Saving POL_pmids_tagged.arff to POL_pmids_tagged (1).arff\n",
            "Saving Polymyalgia Rheumatica_pmids_tagged.arff to Polymyalgia Rheumatica_pmids_tagged (1).arff\n",
            "Saving posterior pituitary_pmids_tagged.arff to posterior pituitary_pmids_tagged (1).arff\n",
            "Saving Potassium_pmids_tagged.arff to Potassium_pmids_tagged (1).arff\n",
            "Saving PR_pmids_tagged.arff to PR_pmids_tagged (1).arff\n",
            "Saving Projection_pmids_tagged.arff to Projection_pmids_tagged (1).arff\n",
            "Saving PVC_pmids_tagged.arff to PVC_pmids_tagged (1).arff\n",
            "Saving RA_pmids_tagged.arff to RA_pmids_tagged (1).arff\n",
            "Saving Radiation_pmids_tagged.arff to Radiation_pmids_tagged (1).arff\n",
            "Saving RB_pmids_tagged.arff to RB_pmids_tagged (1).arff\n",
            "Saving RBC_pmids_tagged.arff to RBC_pmids_tagged (1).arff\n",
            "Saving rDNA_pmids_tagged.arff to rDNA_pmids_tagged (1).arff\n",
            "Saving README.txt to README (1).txt\n",
            "Saving Respiration_pmids_tagged.arff to Respiration_pmids_tagged (1).arff\n",
            "Saving Retinal_pmids_tagged.arff to Retinal_pmids_tagged (1).arff\n",
            "Saving Root_pmids_tagged.arff to Root_pmids_tagged (1).arff\n",
            "Saving RSV_pmids_tagged.arff to RSV_pmids_tagged (1).arff\n",
            "Saving SARS_pmids_tagged.arff to SARS_pmids_tagged (1).arff\n",
            "Saving SARS-associated coronavirus_pmids_tagged.arff to SARS-associated coronavirus_pmids_tagged (1).arff\n",
            "Saving SCD_pmids_tagged.arff to SCD_pmids_tagged (1).arff\n",
            "Saving Schistosoma mansoni_pmids_tagged.arff to Schistosoma mansoni_pmids_tagged (1).arff\n",
            "Saving Semen_pmids_tagged.arff to Semen_pmids_tagged (1).arff\n",
            "Saving sex factor_pmids_tagged.arff to sex factor_pmids_tagged (1).arff\n",
            "Saving SLS_pmids_tagged.arff to SLS_pmids_tagged (1).arff\n",
            "Saving Sodium_pmids_tagged.arff to Sodium_pmids_tagged (1).arff\n",
            "Saving SPR_pmids_tagged.arff to SPR_pmids_tagged (1).arff\n",
            "Saving SS_pmids_tagged.arff to SS_pmids_tagged (1).arff\n",
            "Saving Staph_pmids_tagged.arff to Staph_pmids_tagged (1).arff\n",
            "Saving STEM_pmids_tagged.arff to STEM_pmids_tagged (1).arff\n",
            "Saving Sterilization_pmids_tagged.arff to Sterilization_pmids_tagged (1).arff\n",
            "Saving Strep_pmids_tagged.arff to Strep_pmids_tagged (1).arff\n",
            "Saving Synapsis_pmids_tagged.arff to Synapsis_pmids_tagged (1).arff\n",
            "Saving TAT_pmids_tagged.arff to TAT_pmids_tagged (1).arff\n",
            "Saving Tax_pmids_tagged.arff to Tax_pmids_tagged (1).arff\n",
            "Saving TEM_pmids_tagged.arff to TEM_pmids_tagged (1).arff\n",
            "Saving THYMUS_pmids_tagged.arff to THYMUS_pmids_tagged (1).arff\n",
            "Saving TLC_pmids_tagged.arff to TLC_pmids_tagged (1).arff\n",
            "Saving TMJ_pmids_tagged.arff to TMJ_pmids_tagged (1).arff\n",
            "Saving TMP_pmids_tagged.arff to TMP_pmids_tagged (1).arff\n",
            "Saving TNC_pmids_tagged.arff to TNC_pmids_tagged (1).arff\n",
            "Saving TNT_pmids_tagged.arff to TNT_pmids_tagged (1).arff\n",
            "Saving Tolerance_pmids_tagged.arff to Tolerance_pmids_tagged (1).arff\n",
            "Saving tomography_pmids_tagged.arff to tomography_pmids_tagged (1).arff\n",
            "Saving Torula_pmids_tagged.arff to Torula_pmids_tagged (1).arff\n",
            "Saving TPA_pmids_tagged.arff to TPA_pmids_tagged (1).arff\n",
            "Saving TPO_pmids_tagged.arff to TPO_pmids_tagged (1).arff\n",
            "Saving TRF_pmids_tagged.arff to TRF_pmids_tagged (1).arff\n",
            "Saving TSF_pmids_tagged.arff to TSF_pmids_tagged (1).arff\n",
            "Saving TYR_pmids_tagged.arff to TYR_pmids_tagged (1).arff\n",
            "Saving US_pmids_tagged.arff to US_pmids_tagged (1).arff\n",
            "Saving Ventricles_pmids_tagged.arff to Ventricles_pmids_tagged (1).arff\n",
            "Saving veterinary_pmids_tagged.arff to veterinary_pmids_tagged (1).arff\n",
            "Saving Wasp_pmids_tagged.arff to Wasp_pmids_tagged (1).arff\n",
            "Saving WBS_pmids_tagged.arff to WBS_pmids_tagged (1).arff\n",
            "Saving WT1_pmids_tagged.arff to WT1_pmids_tagged (1).arff\n",
            "Saving Yellow Fever_pmids_tagged.arff to Yellow Fever_pmids_tagged (1).arff\n"
          ]
        },
        {
          "output_type": "error",
          "ename": "AttributeError",
          "evalue": "ignored",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-44-33c5bdac00d1>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m    100\u001b[0m \u001b[0mMSD_Load\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mUploadMSDFiles\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    101\u001b[0m \u001b[0mpaths\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mMSD_Load\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mFilePaths\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/content/'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 102\u001b[0;31m \u001b[0;34m[\u001b[0m\u001b[0mData\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mCUIDic\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mMSD_Load\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mExtractDat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpaths\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    103\u001b[0m \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mMSD_Load\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mString2DF\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mData\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    104\u001b[0m \u001b[0;31m# MSD_Load.WriteTSV(df, 'MSDFile.tsv')\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m<ipython-input-44-33c5bdac00d1>\u001b[0m in \u001b[0;36mExtractDat\u001b[0;34m(Paths)\u001b[0m\n\u001b[1;32m     25\u001b[0m     \u001b[0mCUIDic\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     26\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mentry\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mPaths\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 27\u001b[0;31m       \u001b[0mmatch\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mre\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msearch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mr'/content/(.*)_pmids_tagged.arff'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mentry\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgroup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     28\u001b[0m       \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mentry\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'r'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencoding\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"ISO-8859-1\"\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     29\u001b[0m         \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mAttributeError\u001b[0m: 'NoneType' object has no attribute 'group'"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Many to one examples:\n",
        "# https://towardsdatascience.com/sentiment-analysis-using-lstm-step-by-step-50d074f09948\n",
        "# https://towardsdatascience.com/reading-between-the-layers-lstm-network-7956ad192e58\n",
        "\n",
        "# LSTM Coding tut\n",
        "# https://towardsdatascience.com/lstm-by-example-using-tensorflow-feb0c1968537\n",
        "# https://www.deeplearningbook.org/contents/rnn.html\n",
        "# http://colah.github.io/posts/2015-08-Understanding-LSTMs/\n",
        "\n",
        "# WSD Papers\n",
        "# https://aclanthology.org/C18-1030/\n",
        "# df = MSD_Load.String2DF(Data)\n",
        "# # Clean up data\n",
        "# #   Tokenize? Punct?\n",
        "# #   Standardize lengths\n",
        "# # convert to numbers using frequency\n",
        "# # Generate embeddings\n",
        "# df.head()\n",
        "# print(type(df['citation string'][0]))\n",
        "# print(df['citation string'][0])\n",
        "\n",
        "# for row in df['citation string']:\n",
        "#   for sub_str in ['-1']:\n",
        "#     if sub_str in row:\n",
        "#       print(row)\n",
        "\n",
        "# !pip install glove_python\n",
        "\n",
        "# from glove import Corpus, Glove\n",
        "\n",
        "# #Creating a corpus object\n",
        "# corpus = Corpus() \n",
        "\n",
        "# #Training the corpus to generate the co-occurrence matrix which is used in GloVe\n",
        "# corpus.fit(Corpus, window=10)\n",
        "\n",
        "# glove = Glove(no_components=5, learning_rate=0.05) \n",
        "# glove.fit(corpus.matrix, epochs=30, no_threads=4, verbose=True)\n",
        "# glove.add_dictionary(corpus.dictionary)\n",
        "# glove.save('glove.model')\n",
        "\n",
        "from gensim.models import Word2Vec\n",
        "\n",
        "sg_model = Word2Vec(sentences=Corpus, size=256, window=5, workers=4, sg=1)\n",
        "\n",
        "labels = []\n",
        "tokens = []\n",
        "for word in model.wv.vocab:\n",
        "    tokens.append(model[word])\n",
        "    labels.append(word)"
      ],
      "metadata": {
        "id": "W6P_RLL3yclO"
      },
      "execution_count": 13,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# for word in sg_model.wv.vocab:\n",
        "#   print(word)\n",
        "#   print(sg_model[word])\n",
        "#   break\n",
        "# print(sg_model.wv.vocab[0])\n",
        "print(sg_model['1161'])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 401
        },
        "id": "dyvV6IlIGc46",
        "outputId": "009b85a3-04b7-45f2-a83f-d6fe42d66f85"
      },
      "execution_count": 31,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:6: DeprecationWarning: Call to deprecated `__getitem__` (Method will be removed in 4.0.0, use self.wv.__getitem__() instead).\n",
            "  \n"
          ]
        },
        {
          "output_type": "error",
          "ename": "KeyError",
          "evalue": "ignored",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-31-1951e429c258>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;31m#   break\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;31m# print(sg_model.wv.vocab[0])\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msg_model\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'1161'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/gensim/utils.py\u001b[0m in \u001b[0;36mnew_func1\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m   1420\u001b[0m                     \u001b[0mstacklevel\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1421\u001b[0m                 )\n\u001b[0;32m-> 1422\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1423\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1424\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mnew_func1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/gensim/models/word2vec.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, words)\u001b[0m\n\u001b[1;32m   1101\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1102\u001b[0m         \"\"\"\n\u001b[0;32m-> 1103\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwv\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__getitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mwords\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1104\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1105\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mdeprecated\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Method will be removed in 4.0.0, use self.wv.__contains__() instead\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/gensim/models/keyedvectors.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, entities)\u001b[0m\n\u001b[1;32m    335\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mentities\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstring_types\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    336\u001b[0m             \u001b[0;31m# allow calls like trained_model['office'], as a shorthand for trained_model[['office']]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 337\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_vector\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mentities\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    338\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    339\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mvstack\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_vector\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mentity\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mentity\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mentities\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/gensim/models/keyedvectors.py\u001b[0m in \u001b[0;36mget_vector\u001b[0;34m(self, word)\u001b[0m\n\u001b[1;32m    453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mget_vector\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mword\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 455\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mword_vec\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mword\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    456\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    457\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mwords_closer_than\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mw1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mw2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/gensim/models/keyedvectors.py\u001b[0m in \u001b[0;36mword_vec\u001b[0;34m(self, word, use_norm)\u001b[0m\n\u001b[1;32m    450\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    451\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 452\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"word '%s' not in vocabulary\"\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mword\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mget_vector\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mword\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mKeyError\u001b[0m: \"word '1161' not in vocabulary\""
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# import nltk\n",
        "\n",
        "# word_data = \"It's originated from the idea that there are readers who prefer learning new skills from the comforts of their drawing rooms\"\n",
        "# nltk_tokens = nltk.word_tokenize(word_data)\n",
        "# print (nltk_tokens)\n",
        "import nltk\n",
        "import string\n",
        "# nltk.download('punkt')\n",
        "nltk.download(\"stopwords\")\n",
        "from nltk.corpus import stopwords\n",
        "\n",
        "\n",
        "from nltk.tokenize import word_tokenize\n",
        "s = df['citation string'][0]\n",
        "# print(dir(s))\n",
        "df.head()\n",
        "# s = word_tokenize(s.lower())\n",
        "print(s)\n",
        "print(string.punctuation)\n",
        "print(stopwords.words('english'))\n",
        "test = df\n",
        "# test['citation string'][0] = word_tokenize(test['citation string'][0].lower())\n",
        "test.head()\n",
        "# print(test['citation string'][0])"
      ],
      "metadata": {
        "id": "oH1wtXE75M02"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "'''\n",
        "Data processing\n",
        "'''\n",
        "# Convert everything to lower\n",
        "# remove punctuation\n",
        "#   Remove all but <>\n",
        "#   Replace tag with $ and then remove <>\n",
        "# Tokenize the string\n",
        "# Remove stop words\n",
        "# Return data?\n",
        "#   truncating or removing data?\n",
        "\n",
        "import nltk\n",
        "import string\n",
        "nltk.download('punkt')\n",
        "nltk.download(\"stopwords\")\n",
        "from nltk.corpus import stopwords\n",
        "from nltk.tokenize import word_tokenize\n",
        "\n",
        "class Data_Processing:\n",
        "\n",
        "  # Basic tokenization\n",
        "  # Note: consider adding start and end sent tokens? doesn't seeme relecant to learning word context\n",
        "  def BasicProc(df):\n",
        "    corpus = []\n",
        "    labels = []\n",
        "    for index, row in df.iterrows():\n",
        "      # Remove punctuation and replace target word with $\n",
        "      punctuation = string.punctuation.replace('<', '')\n",
        "      punctuation = punctuation.replace('>', '')\n",
        "      row['citation string'] = row['citation string'].translate(str.maketrans('','',punctuation))\n",
        "      subString = re.search(r'<e>(.*)<e>',row['citation string']).group(1)\n",
        "      row['citation string'] = row['citation string'].replace('<e>' + subString + '<e>', '$')\n",
        "      row['citation string'] = row['citation string'].translate(str.maketrans('','','<>'))\n",
        "\n",
        "      # Lower, tokenize, and remove stop words\n",
        "      row['citation string'] = word_tokenize(row['citation string'].lower())\n",
        "      filteredSentence = []\n",
        "      stopWords = stopwords.words('english')\n",
        "      for word in row['citation string']:\n",
        "        if word not in stopWords:\n",
        "          filteredSentence.append(word)\n",
        "      row['citation string'] = filteredSentence\n",
        "\n",
        "      # Update the df\n",
        "      corpus.append(row['citation string'])\n",
        "      labels.append(row['CUI'])\n",
        "    return corpus, labels\n",
        "  \n",
        "  def DataSnip(Corpus, Window):\n",
        "    WindowedCorpus = []\n",
        "    for entry in Corpus:\n",
        "      \n",
        "      # Add Window # of blanks to start and end to guarantee correct lengths\n",
        "      Blanks = [' '] * Window   ##################3\n",
        "      entry = Blanks + entry + Blanks\n",
        "\n",
        "      # Find relevant segment\n",
        "      index = 0\n",
        "      for token in entry:\n",
        "        if token == '$':\n",
        "          break\n",
        "        else:\n",
        "          index += 1\n",
        "\n",
        "      WindowedCorpus.append(entry[index-Window:Window+index+1])\n",
        "    return WindowedCorpus\n",
        "\n",
        "WindowSize = 10\n",
        "[Corpus, Labels] = Data_Processing.BasicProc(df)\n",
        "WindowedCorpus = Data_Processing.DataSnip(Corpus, WindowSize)"
      ],
      "metadata": {
        "id": "kPDzsjSfNiGH",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0075cad1-04cf-4dfb-ce0a-7bee7359a561"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "[nltk_data] Downloading package punkt to /root/nltk_data...\n",
            "[nltk_data]   Unzipping tokenizers/punkt.zip.\n",
            "[nltk_data] Downloading package stopwords to /root/nltk_data...\n",
            "[nltk_data]   Unzipping corpora/stopwords.zip.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "'''\n",
        "Embeddings\n",
        "'''\n",
        "\n",
        "import numpy as np\n",
        "from gensim.models import Word2Vec\n",
        "\n",
        "class Embedding():\n",
        "\n",
        "  def RevFreq(Corpus):\n",
        "    tokenKey = {}\n",
        "    for element in Corpus:\n",
        "      for token in element:\n",
        "        if token in tokenKey:\n",
        "          tokenKey[token] += 1\n",
        "        else:\n",
        "          tokenKey[token] = 1\n",
        "\n",
        "    sortedResCounts = sorted(tokenKey.items(), key=lambda x:x[1], reverse=True)  # Tuples\n",
        "    Count = 1\n",
        "    Zeros = ['', '$']   # The $ tag and empty spaces are special cases not relevant to the surrounding context\n",
        "    sortedRes = {}\n",
        "    for key in sortedResCounts:\n",
        "      if key[0] in Zeros:\n",
        "        sortedRes[key[0]] = [0]\n",
        "      else:\n",
        "        sortedRes[key[0]] = [Count]\n",
        "        Count +=1\n",
        "\n",
        "    return sortedRes\n",
        "\n",
        "  # def Word2Vec(Corpus):\n",
        "  #   sg_model = Word2Vec(sentences=Corpus, size=256, window=5, workers=4, sg=1)\n",
        "\n",
        "  #   # For some reason this is skipping words like 1161\n",
        "  #   sg_model_dict = {}\n",
        "  #   for word in sg_model.wv.vocab:\n",
        "  #     sg_model_dict[word] = sg_model[word]\n",
        "\n",
        "  #   return sg_model_dict\n",
        "\n",
        "  # def Glove():\n",
        "    # Glove\n",
        "    # word2vec\n",
        "    # ?\n",
        "\n",
        "  def OneHotEncoding(Labels):\n",
        "    CompressedLabels = [*set(Labels)]\n",
        "    Dim = len(CompressedLabels)\n",
        "    OneHotMat = np.identity(Dim)\n",
        "    LabelDict = {}\n",
        "    for i in range(0,Dim):\n",
        "      Blank = []\n",
        "      for element in OneHotMat[i]:\n",
        "        Blank.append([element])\n",
        "\n",
        "      LabelDict[CompressedLabels[i]] = Blank\n",
        "\n",
        "    return LabelDict\n",
        "  \n",
        "  def DictConvertion(List, Dict, Forward = True):\n",
        "    if Forward is True:\n",
        "      ConvertedList = []\n",
        "      for element in List:\n",
        "        ConvertedList.append(Dict[element])\n",
        "    else:\n",
        "      ConvertedList = []\n",
        "      for element in List:\n",
        "        ConvertedList.append( list(Dict.keys())[list(Dict.values()).index(element)])\n",
        "\n",
        "    return ConvertedList\n",
        "\n",
        "# Keys\n",
        "TokenKey = Embedding.RevFreq(WindowedCorpus)\n",
        "# TokenKey = Embedding.Word2Vec(Corpus)\n",
        "OneHotLabels = Embedding.OneHotEncoding(Labels)\n",
        "\n",
        "# Convert tokens and labels using keys\n",
        "ConvertedLabels = Embedding.DictConvertion(Labels, OneHotLabels)\n",
        "ConvertedCorpus = []\n",
        "for element in WindowedCorpus:\n",
        "  ConvertedCorpus.append(Embedding.DictConvertion(element, TokenKey))"
      ],
      "metadata": {
        "id": "-hl0WqQgw6bN",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "d6dbe7ef-a7e5-47ae-c27d-50e4a5e44aac"
      },
      "execution_count": 45,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[1;30;43mStreaming output truncated to the last 5000 lines.\u001b[0m\n",
            "standard\n",
            "diet\n",
            "modulate\n",
            "ion\n",
            "currents\n",
            "emerging\n",
            "evidences\n",
            "point\n",
            "similar\n",
            "effects\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "vegetable\n",
            "n3\n",
            "pufa\n",
            "much\n",
            "less\n",
            "known\n",
            "effects\n",
            "specific\n",
            "cardiac\n",
            "ion\n",
            "tail\n",
            "first\n",
            "mode\n",
            "fatty\n",
            "acid\n",
            "binding\n",
            "sitedirected\n",
            "mutagenesis\n",
            "active\n",
            "site\n",
            "$\n",
            "ala215\n",
            "typically\n",
            "conserved\n",
            "gly\n",
            "rlox\n",
            "revealed\n",
            "substitution\n",
            "gly\n",
            "retained\n",
            "9r\n",
            "acidenriched\n",
            "oil\n",
            "ace\n",
            "inhibitor\n",
            "concerning\n",
            "decrease\n",
            "blood\n",
            "pressure\n",
            "shralphalinolenic\n",
            "acid\n",
            "$\n",
            "reported\n",
            "exhibit\n",
            "antihypertensive\n",
            "effect\n",
            "angiotensinconverting\n",
            "enzyme\n",
            "inhibitor\n",
            "acei\n",
            "also\n",
            "antihypertensive\n",
            "examined\n",
            "influence\n",
            "17betaestradiol\n",
            "e2\n",
            "capacity\n",
            "shsy5y\n",
            "cells\n",
            "supplemented\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "produce\n",
            "eicosapentaenoic\n",
            "acid\n",
            "epa\n",
            "docosapentaenoic\n",
            "acid\n",
            "dpa\n",
            "docosahexaenoic\n",
            "acid\n",
            "dha\n",
            "sunflower\n",
            "differ\n",
            "greatly\n",
            "content\n",
            "n3\n",
            "fatty\n",
            "acids\n",
            "specifically\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "low\n",
            "ala\n",
            "intake\n",
            "associated\n",
            "risk\n",
            "fatal\n",
            "chd\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "administrationthe\n",
            "present\n",
            "study\n",
            "attempted\n",
            "clarify\n",
            "antihypertensive\n",
            "effect\n",
            "mechanism\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "administered\n",
            "orally\n",
            "purpose\n",
            "1\n",
            "ml\n",
            "flaxseed\n",
            "oil\n",
            "rich\n",
            "ala\n",
            "high\n",
            "blood\n",
            "pressure\n",
            "mild\n",
            "hypertensionwe\n",
            "investigated\n",
            "antihypertensive\n",
            "effect\n",
            "safety\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "human\n",
            "subjects\n",
            "experiment\n",
            "1\n",
            "subjects\n",
            "highnormal\n",
            "blood\n",
            "pressure\n",
            "mild\n",
            "hypertension\n",
            "gammalinolenic\n",
            "acid\n",
            "gla\n",
            "gammalinolenic\n",
            "acid\n",
            "gla\n",
            "183cisa6912\n",
            "6\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "183cisdelta9\n",
            "12\n",
            "15\n",
            "omega3product\n",
            "dehydrogenation\n",
            "linoleic\n",
            "acid\n",
            "la\n",
            "important\n",
            "efa\n",
            "frequent\n",
            "sequela\n",
            "type\n",
            "2\n",
            "diabetes\n",
            "epidemic\n",
            "around\n",
            "world\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "183\n",
            "n3\n",
            "eicosapentaenoic\n",
            "acid\n",
            "epa\n",
            "205\n",
            "n3\n",
            "docosahexaenoic\n",
            "acid\n",
            "dha\n",
            "adultsour\n",
            "first\n",
            "aim\n",
            "determine\n",
            "whether\n",
            "isocaloric\n",
            "intervention\n",
            "using\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "form\n",
            "flaxseed\n",
            "oil\n",
            "would\n",
            "alter\n",
            "adiponectin\n",
            "levels\n",
            "among\n",
            "overweight\n",
            "otherwise\n",
            "endothelial\n",
            "cells\n",
            "contributes\n",
            "acceleration\n",
            "atherosclerosis\n",
            "associated\n",
            "diabetes\n",
            "hypothesized\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "might\n",
            "attenuate\n",
            "high\n",
            "glucoseinduced\n",
            "apoptosis\n",
            "cultured\n",
            "human\n",
            "umbilical\n",
            "vein\n",
            "endothelial\n",
            "oil\n",
            "supplementation\n",
            "plasma\n",
            "adiponectin\n",
            "levels\n",
            "dyslipidemic\n",
            "menbackground\n",
            "dietary\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "associated\n",
            "reduced\n",
            "risk\n",
            "development\n",
            "atherosclerosis\n",
            "adiponectin\n",
            "hormone\n",
            "specifically\n",
            "secreted\n",
            "adipocytes\n",
            "gland\n",
            "promoter\n",
            "found\n",
            "produce\n",
            "milk\n",
            "containing\n",
            "significantly\n",
            "elevated\n",
            "levels\n",
            "alphalinolenic\n",
            "$\n",
            "eicosapentaenoic\n",
            "epa\n",
            "acid\n",
            "study\n",
            "preweaning\n",
            "growth\n",
            "rate\n",
            "development\n",
            "mouse\n",
            "pups\n",
            "isocaloric\n",
            "diet\n",
            "supplemented\n",
            "plantbased\n",
            "dietary\n",
            "omega3\n",
            "fatty\n",
            "acid\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "interleukin6\n",
            "creactive\n",
            "protein\n",
            "serum\n",
            "amyloid\n",
            "tumor\n",
            "necrosis\n",
            "factoralpha\n",
            "subjects\n",
            "included\n",
            "surrounds\n",
            "extent\n",
            "conversion\n",
            "humans\n",
            "parent\n",
            "pufa\n",
            "linoleic\n",
            "acid\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "respective\n",
            "longchain\n",
            "pufa\n",
            "products\n",
            "result\n",
            "uncertainty\n",
            "potential\n",
            "benefits\n",
            "ala\n",
            "versus\n",
            "diet\n",
            "supplemented\n",
            "highlinoleic\n",
            "acid\n",
            "lna\n",
            "omega6\n",
            "safflower\n",
            "oil\n",
            "highalphalinolenic\n",
            "acid\n",
            "$\n",
            "omega3\n",
            "perilla\n",
            "oil\n",
            "long\n",
            "periods\n",
            "results\n",
            "suggested\n",
            "perilla\n",
            "oil\n",
            "low\n",
            "study\n",
            "examine\n",
            "effect\n",
            "moderate\n",
            "intakes\n",
            "associated\n",
            "recommended\n",
            "intake\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "lcat\n",
            "activity\n",
            "humans\n",
            "two\n",
            "experimental\n",
            "diets\n",
            "tested\n",
            "3\n",
            "months\n",
            "diet\n",
            "clear\n",
            "zone\n",
            "scheimpflug\n",
            "slit\n",
            "image\n",
            "lens\n",
            "results\n",
            "intake\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "positively\n",
            "associated\n",
            "change\n",
            "nuclear\n",
            "density\n",
            "geometric\n",
            "mean\n",
            "nuclear\n",
            "density\n",
            "change\n",
            "synthase\n",
            "gene\n",
            "expression\n",
            "via\n",
            "nfkappab\n",
            "mitogenactivated\n",
            "protein\n",
            "kinase\n",
            "pathwaysalphalinolenic\n",
            "acid\n",
            "$\n",
            "isolated\n",
            "actinidia\n",
            "polygama\n",
            "fruits\n",
            "exhibits\n",
            "potent\n",
            "antiinflammatory\n",
            "activity\n",
            "unknown\n",
            "mechanism\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "competition\n",
            "245n3\n",
            "$\n",
            "delta\n",
            "6\n",
            "desaturase\n",
            "may\n",
            "limit\n",
            "accumulation\n",
            "dha\n",
            "hepg2\n",
            "cell\n",
            "membranesthe\n",
            "present\n",
            "study\n",
            "designed\n",
            "evaluate\n",
            "whether\n",
            "longterm\n",
            "intake\n",
            "dietary\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "supplied\n",
            "whole\n",
            "grainextruded\n",
            "linseed\n",
            "increase\n",
            "endogenous\n",
            "production\n",
            "n3\n",
            "longchain\n",
            "polyunsaturated\n",
            "fads2\n",
            "affects\n",
            "pufa\n",
            "biosynthetic\n",
            "pathway\n",
            "consequently\n",
            "modifies\n",
            "effect\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "myocardial\n",
            "infarction\n",
            "mi\n",
            "design\n",
            "case\n",
            "subjects\n",
            "n\n",
            "1694\n",
            "first\n",
            "nonfatal\n",
            "healthy\n",
            "menhigh\n",
            "linoleic\n",
            "acid\n",
            "la\n",
            "intakes\n",
            "suggested\n",
            "reduce\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "183n3\n",
            "metabolism\n",
            "eicosapentaenoic\n",
            "acid\n",
            "epa\n",
            "205n3\n",
            "docosahexaenoic\n",
            "acid\n",
            "dha\n",
            "226n3\n",
            "human\n",
            "volunteersa\n",
            "ratelimiting\n",
            "step\n",
            "docosahexaenoic\n",
            "acid\n",
            "dha\n",
            "formation\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "involves\n",
            "peroxisomal\n",
            "oxidation\n",
            "246n3\n",
            "dha\n",
            "aim\n",
            "study\n",
            "determine\n",
            "whether\n",
            "conjugated\n",
            "atherosclerosis\n",
            "chronic\n",
            "inflammatory\n",
            "disease\n",
            "previously\n",
            "reported\n",
            "diet\n",
            "high\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "reduces\n",
            "lipid\n",
            "inflammatory\n",
            "cardiovascular\n",
            "disease\n",
            "risk\n",
            "factors\n",
            "hypercholesterolemic\n",
            "subjects\n",
            "objective\n",
            "specific\n",
            "needs\n",
            "women\n",
            "omega3\n",
            "fatty\n",
            "acids\n",
            "including\n",
            "alpha\n",
            "linoleic\n",
            "acid\n",
            "$\n",
            "long\n",
            "chain\n",
            "fatty\n",
            "acids\n",
            "eicosapentaenoic\n",
            "acid\n",
            "epa\n",
            "docosahexaenoic\n",
            "acid\n",
            "dha\n",
            "supplementation\n",
            "flaxseed\n",
            "oil\n",
            "lowers\n",
            "blood\n",
            "pressure\n",
            "dyslipidaemic\n",
            "patientsobjective\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "natural\n",
            "precursor\n",
            "cardioprotective\n",
            "longchain\n",
            "n3\n",
            "fatty\n",
            "acids\n",
            "available\n",
            "data\n",
            "indicate\n",
            "micewe\n",
            "recently\n",
            "suggested\n",
            "feeding\n",
            "c18\n",
            "polyunsaturated\n",
            "fatty\n",
            "acid\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "instead\n",
            "linoleic\n",
            "acid\n",
            "la\n",
            "reduced\n",
            "body\n",
            "fat\n",
            "mice\n",
            "however\n",
            "difference\n",
            "13\n",
            "monounsaturated\n",
            "fatty\n",
            "acids\n",
            "mufa\n",
            "9\n",
            "pufa\n",
            "77\n",
            "la\n",
            "08\n",
            "$\n",
            "2\n",
            "linoleic\n",
            "acid\n",
            "diet\n",
            "la\n",
            "37\n",
            "total\n",
            "fat\n",
            "9\n",
            "sfa\n",
            "present\n",
            "study\n",
            "deltasarcoglycannull\n",
            "cardiomyopathic\n",
            "hamsters\n",
            "fed\n",
            "weaning\n",
            "death\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "enriched\n",
            "versus\n",
            "standard\n",
            "diet\n",
            "results\n",
            "demonstrated\n",
            "great\n",
            "accumulation\n",
            "ala\n",
            "eicosapentaenoic\n",
            "carbon\n",
            "recycling\n",
            "13calphalinolenateunder\n",
            "various\n",
            "dietary\n",
            "physiological\n",
            "conditions\n",
            "carbon\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "extensively\n",
            "recycled\n",
            "saturated\n",
            "monounsaturated\n",
            "fatty\n",
            "acids\n",
            "study\n",
            "investigated\n",
            "whether\n",
            "carbon\n",
            "\n",
            "her2\n",
            "erbb2targeted\n",
            "effects\n",
            "omega3\n",
            "polyunsaturated\n",
            "fatty\n",
            "acid\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "183n3\n",
            "breast\n",
            "cancer\n",
            "cells\n",
            "fat\n",
            "features\n",
            "mediterranean\n",
            "diet\n",
            "antiher2\n",
            "cocktail\n",
            "hempseed\n",
            "contains\n",
            "high\n",
            "proportion\n",
            "pufas\n",
            "linoleic\n",
            "acid\n",
            "la\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "may\n",
            "opposing\n",
            "effects\n",
            "postischemic\n",
            "heart\n",
            "performance\n",
            "reported\n",
            "data\n",
            "concerning\n",
            "cardiovascular\n",
            "deathbackground\n",
            "several\n",
            "animal\n",
            "experiments\n",
            "documented\n",
            "omega3\n",
            "fatty\n",
            "acid\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "c\n",
            "183\n",
            "n3\n",
            "vegetable\n",
            "oils\n",
            "beneficial\n",
            "electrophysiological\n",
            "antiarrhythmic\n",
            "effects\n",
            "may\n",
            "dietary\n",
            "intake\n",
            "omega3\n",
            "polyunsaturated\n",
            "fatty\n",
            "acids\n",
            "pufa\n",
            "like\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "antiarrhythmic\n",
            "cardioprotective\n",
            "pufa\n",
            "may\n",
            "also\n",
            "beneficial\n",
            "hypertension\n",
            "altered\n",
            "naca2\n",
            "exchanger\n",
            "disease\n",
            "chd\n",
            "striking\n",
            "protective\n",
            "effect\n",
            "mediterranean\n",
            "diet\n",
            "rich\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "reported\n",
            "lyon\n",
            "diet\n",
            "heart\n",
            "study\n",
            "50\n",
            "70\n",
            "reduction\n",
            "risk\n",
            "recurrence\n",
            "supplementation\n",
            "affect\n",
            "plasma\n",
            "lipoprotein\n",
            "concentration\n",
            "particle\n",
            "size\n",
            "human\n",
            "subjectsalphalinolenic\n",
            "acid\n",
            "$\n",
            "major\n",
            "dietary\n",
            "n3\n",
            "fatty\n",
            "acid\n",
            "clinical\n",
            "trials\n",
            "ala\n",
            "supplementation\n",
            "shown\n",
            "docosahexaenoic\n",
            "acid\n",
            "dha\n",
            "eicosapentaenoic\n",
            "acid\n",
            "epa\n",
            "biochemical\n",
            "precursor\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "actually\n",
            "ala\n",
            "beneficial\n",
            "effects\n",
            "apart\n",
            "known\n",
            "antiarrhythmic\n",
            "effect\n",
            "fact\n",
            "ala\n",
            "intake\n",
            "four\n",
            "walnuts\n",
            "day\n",
            "humansbackground\n",
            "aims\n",
            "ingestion\n",
            "alpha\n",
            "linolenic\n",
            "acid\n",
            "$\n",
            "richest\n",
            "source\n",
            "among\n",
            "dry\n",
            "fruits\n",
            "walnuts\n",
            "associated\n",
            "cardiovascular\n",
            "prevention\n",
            "aim\n",
            "intake\n",
            "eicosapentaenoic\n",
            "acid\n",
            "epa\n",
            "docosahexaenoic\n",
            "acid\n",
            "dha\n",
            "cases\n",
            "alphalinolenic\n",
            "acid\n",
            "$\n",
            "associated\n",
            "reduced\n",
            "risk\n",
            "cardiovascular\n",
            "events\n",
            "death\n",
            "association\n",
            "n3\n",
            "fatty\n",
            "acids\n",
            "activities\n",
            "also\n",
            "associated\n",
            "relatively\n",
            "high\n",
            "risk\n",
            "death\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "frequent\n",
            "cause\n",
            "nontraumatic\n",
            "death\n",
            "males\n",
            "older\n",
            "34\n",
            "years\n",
            "altitude\n",
            "leisure\n",
            "absence\n",
            "underlying\n",
            "cardiac\n",
            "disease\n",
            "therefore\n",
            "pathologic\n",
            "cardiac\n",
            "repolarization\n",
            "could\n",
            "promote\n",
            "$\n",
            "people\n",
            "epilepsy\n",
            "could\n",
            "one\n",
            "plausible\n",
            "cause\n",
            "sudep\n",
            "herein\n",
            "review\n",
            "abnormal\n",
            "hypertrophic\n",
            "obstructive\n",
            "cardiomyopathy\n",
            "hocm\n",
            "acknowledged\n",
            "risk\n",
            "factor\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "otherwise\n",
            "healthy\n",
            "active\n",
            "patient\n",
            "incidence\n",
            "scd\n",
            "young\n",
            "people\n",
            "high\n",
            "enough\n",
            "ii\n",
            "patients\n",
            "implanted\n",
            "cardioverter\n",
            "defibrillatorthe\n",
            "evadef\n",
            "studybackground\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "predominant\n",
            "cause\n",
            "mortality\n",
            "patients\n",
            "mild\n",
            "heart\n",
            "failure\n",
            "hf\n",
            "2year\n",
            "followup\n",
            "baseline\n",
            "cystatin\n",
            "c\n",
            "creatinine\n",
            "measured\n",
            "baseline\n",
            "sera\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "defined\n",
            "sudden\n",
            "pulseless\n",
            "condition\n",
            "cardiac\n",
            "origin\n",
            "previously\n",
            "stable\n",
            "individual\n",
            "occurred\n",
            "children\n",
            "attention\n",
            "deficithyperactivity\n",
            "disorder\n",
            "may\n",
            "increase\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "examined\n",
            "costeffectiveness\n",
            "pretreatment\n",
            "screening\n",
            "ecg\n",
            "reducing\n",
            "scd\n",
            "risk\n",
            "children\n",
            "diagnosed\n",
            "antipsychotics\n",
            "conventional\n",
            "atypical\n",
            "tca\n",
            "thioridazine\n",
            "haloperidol\n",
            "shared\n",
            "similar\n",
            "relative\n",
            "risk\n",
            "$\n",
            "lower\n",
            "doses\n",
            "risperidone\n",
            "higher\n",
            "relative\n",
            "risk\n",
            "haloperidol\n",
            "cardiac\n",
            "arrest\n",
            "ventricular\n",
            "n3\n",
            "polyunsaturated\n",
            "fatty\n",
            "acids\n",
            "pufa\n",
            "offer\n",
            "protection\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "autonomic\n",
            "nervous\n",
            "system\n",
            "involved\n",
            "pathogenesis\n",
            "scd\n",
            "due\n",
            "fact\n",
            "n3\n",
            "pufa\n",
            "heart\n",
            "failure\n",
            "conflicting\n",
            "examined\n",
            "risk\n",
            "allcause\n",
            "mortality\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "1231\n",
            "patients\n",
            "myocardial\n",
            "infarction\n",
            "left\n",
            "ventricular\n",
            "dysfunction\n",
            "enrolled\n",
            "multicenter\n",
            "automatic\n",
            "\n",
            "\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "epidemiology\n",
            "risk\n",
            "factorssudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "important\n",
            "publichealth\n",
            "problem\n",
            "multiple\n",
            "etiologies\n",
            "risk\n",
            "factors\n",
            "changing\n",
            "temporal\n",
            "trends\n",
            "implantable\n",
            "cardioverterdefibrillator\n",
            "icd\n",
            "therapy\n",
            "established\n",
            "intervention\n",
            "prevention\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "patients\n",
            "significant\n",
            "left\n",
            "ventricular\n",
            "dysfunction\n",
            "multiple\n",
            "randomized\n",
            "clinical\n",
            "trials\n",
            "studied\n",
            "death\n",
            "young\n",
            "competitive\n",
            "athletes\n",
            "due\n",
            "genetic\n",
            "cardiac\n",
            "abnormalitiessudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "young\n",
            "athletes\n",
            "generally\n",
            "caused\n",
            "inherited\n",
            "cardiac\n",
            "disorders\n",
            "events\n",
            "relatively\n",
            "compared\n",
            "review\n",
            "recommended\n",
            "major\n",
            "possible\n",
            "clinical\n",
            "risk\n",
            "markers\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "hypertrophic\n",
            "cardiomyopathy\n",
            "hcm\n",
            "searched\n",
            "medline\n",
            "embase\n",
            "cochrane\n",
            "databases\n",
            "articles\n",
            "published\n",
            "management\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "coronary\n",
            "artery\n",
            "diseasemanagement\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "undergoing\n",
            "radical\n",
            "change\n",
            "direction\n",
            "becoming\n",
            "increasingly\n",
            "appreciated\n",
            "besides\n",
            "depressed\n",
            "left\n",
            "anatomical\n",
            "metabolic\n",
            "genetic\n",
            "evaluationbackground\n",
            "identifying\n",
            "persons\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "challenging\n",
            "comprehensive\n",
            "evaluation\n",
            "may\n",
            "reveal\n",
            "clues\n",
            "clinical\n",
            "anatomical\n",
            "genetic\n",
            "metabolic\n",
            "right\n",
            "ventricular\n",
            "cardiomyopathy\n",
            "arvcm\n",
            "carry\n",
            "high\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "ventricular\n",
            "tachyarrhythmias\n",
            "often\n",
            "cause\n",
            "scd\n",
            "treated\n",
            "internal\n",
            "cardioverter\n",
            "defibrillators\n",
            "icds\n",
            "patients\n",
            "diagnosed\n",
            "electrical\n",
            "cardiomyopathy\n",
            "increased\n",
            "risk\n",
            "syncope\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "usually\n",
            "dealing\n",
            "young\n",
            "patients\n",
            "even\n",
            "children\n",
            "prevalence\n",
            "diseases\n",
            "low\n",
            "large\n",
            "ami\n",
            "patients\n",
            "decision\n",
            "making\n",
            "regarding\n",
            "primary\n",
            "prevention\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "total\n",
            "676\n",
            "consecutive\n",
            "ami\n",
            "patients\n",
            "78\n",
            "male\n",
            "mean\n",
            "age\n",
            "59\n",
            "families\n",
            "discussion\n",
            "practical\n",
            "legal\n",
            "ethical\n",
            "aspects\n",
            "multidisciplinary\n",
            "collaborationsudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "major\n",
            "cause\n",
            "premature\n",
            "death\n",
            "young\n",
            "adults\n",
            "children\n",
            "developed\n",
            "countries\n",
            "standard\n",
            "45\n",
            "yearsaim\n",
            "study\n",
            "determination\n",
            "incidence\n",
            "structure\n",
            "sudden\n",
            "hospital\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "subjects\n",
            "aged\n",
            "145\n",
            "years\n",
            "moscow\n",
            "analyzed\n",
            "rate\n",
            "structure\n",
            "sd\n",
            "among\n",
            "voltage\n",
            "therapy\n",
            "preventing\n",
            "treating\n",
            "asystole\n",
            "pea\n",
            "icdsintroduction\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "takes\n",
            "500000\n",
            "lives\n",
            "year\n",
            "victim\n",
            "even\n",
            "treated\n",
            "address\n",
            "implantable\n",
            "cardioverter\n",
            "outofhospital\n",
            "patient\n",
            "serious\n",
            "arrhythmia\n",
            "andor\n",
            "possesses\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "system\n",
            "features\n",
            "realtime\n",
            "continuous\n",
            "telemetry\n",
            "monitoring\n",
            "wide\n",
            "range\n",
            "long\n",
            "distance\n",
            "may\n",
            "predispose\n",
            "patients\n",
            "electrophysiologic\n",
            "instability\n",
            "ventricular\n",
            "arrhythmias\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "studied\n",
            "330\n",
            "patients\n",
            "hc\n",
            "judged\n",
            "clinically\n",
            "high\n",
            "risk\n",
            "scd\n",
            "therefore\n",
            "gt75\n",
            "narrowing\n",
            "coronary\n",
            "artery\n",
            "diseasethe\n",
            "currently\n",
            "available\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "risk\n",
            "prediction\n",
            "tools\n",
            "fail\n",
            "identify\n",
            "atrisk\n",
            "patients\n",
            "delineate\n",
            "specific\n",
            "patients\n",
            "myocardial\n",
            "infarctiondespite\n",
            "decline\n",
            "overall\n",
            "cardiovascular\n",
            "mortality\n",
            "incidence\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "continues\n",
            "rise\n",
            "patients\n",
            "survive\n",
            "myocardial\n",
            "infarction\n",
            "mi\n",
            "depressed\n",
            "ejection\n",
            "fraction\n",
            "could\n",
            "identify\n",
            "heart\n",
            "failure\n",
            "hf\n",
            "patients\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "background\n",
            "prediction\n",
            "scd\n",
            "remains\n",
            "important\n",
            "challenge\n",
            "patients\n",
            "mildtomoderate\n",
            "chronic\n",
            "hf\n",
            "5year\n",
            "mortality\n",
            "rate\n",
            "ranging\n",
            "14\n",
            "50\n",
            "owing\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "pump\n",
            "failure\n",
            "risk\n",
            "factors\n",
            "scd\n",
            "adults\n",
            "idc\n",
            "include\n",
            "asymptomatic\n",
            "nonsustained\n",
            "death\n",
            "heart\n",
            "failure\n",
            "trialbackground\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "heart\n",
            "failure\n",
            "trial\n",
            "$\n",
            "heft\n",
            "demonstrated\n",
            "implantable\n",
            "cardioverterdefibrillator\n",
            "icd\n",
            "therapy\n",
            "reduces\n",
            "allcause\n",
            "mortality\n",
            "patients\n",
            "heartrate\n",
            "variability\n",
            "hrv\n",
            "improving\n",
            "risk\n",
            "stratification\n",
            "predict\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "patients\n",
            "established\n",
            "ischemicdilated\n",
            "cardiomyopathy\n",
            "idc\n",
            "shortterm\n",
            "hrv\n",
            "analyzed\n",
            "daytime\n",
            "nighttime\n",
            "sudden\n",
            "arrhythmic\n",
            "cardiac\n",
            "deathbackground\n",
            "although\n",
            "heritable\n",
            "basis\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "suggested\n",
            "impact\n",
            "family\n",
            "history\n",
            "scd\n",
            "risk\n",
            "common\n",
            "genetic\n",
            "determinants\n",
            "difficult\n",
            "indicate\n",
            "genetic\n",
            "factors\n",
            "important\n",
            "contributors\n",
            "determining\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "recent\n",
            "evidence\n",
            "indicates\n",
            "common\n",
            "genetic\n",
            "variants\n",
            "several\n",
            "genes\n",
            "explain\n",
            "part\n",
            "recordingaims\n",
            "high\n",
            "resting\n",
            "heart\n",
            "rate\n",
            "hr\n",
            "associated\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "association\n",
            "fully\n",
            "explained\n",
            "reported\n",
            "association\n",
            "hr\n",
            "coronary\n",
            "heart\n",
            "disease\n",
            "chd\n",
            "review\n",
            "several\n",
            "invasive\n",
            "noninvasive\n",
            "tests\n",
            "risk\n",
            "stratification\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "studied\n",
            "tests\n",
            "microwave\n",
            "wave\n",
            "alternans\n",
            "repolarization\n",
            "abnormality\n",
            "signalaveraged\n",
            "ecg\n",
            "depolarization\n",
            "cv\n",
            "hospitalizationcv\n",
            "mortality\n",
            "24\n",
            "p\n",
            "lt\n",
            "00001\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "34\n",
            "p\n",
            "lt\n",
            "00001\n",
            "contrast\n",
            "later\n",
            "eplerenone\n",
            "initiation\n",
            "gt\n",
            "7\n",
            "heart\n",
            "failureaims\n",
            "assess\n",
            "involvement\n",
            "sleep\n",
            "apnoea\n",
            "nocturnal\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "evaluating\n",
            "cardiac\n",
            "electrical\n",
            "instability\n",
            "using\n",
            "twave\n",
            "alternans\n",
            "twa\n",
            "risk\n",
            "marker\n",
            "isomers\n",
            "may\n",
            "different\n",
            "effects\n",
            "risk\n",
            "particularly\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "methods\n",
            "prospectively\n",
            "examined\n",
            "association\n",
            "dietary\n",
            "intake\n",
            "trans\n",
            "fat\n",
            "scd\n",
            "among\n",
            "associated\n",
            "genetic\n",
            "testing\n",
            "severe\n",
            "symptoms\n",
            "ventricular\n",
            "arrhythmias\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "methods\n",
            "results\n",
            "four\n",
            "hundred\n",
            "twentyfour\n",
            "patients\n",
            "hcm\n",
            "age5516\n",
            "years\n",
            "range\n",
            "myocardial\n",
            "perfusion\n",
            "imaging\n",
            "identify\n",
            "patients\n",
            "increased\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "methods\n",
            "results\n",
            "analyzed\n",
            "cohort\n",
            "6383\n",
            "patients\n",
            "angiographically\n",
            "documented\n",
            "coronary\n",
            "artery\n",
            "used\n",
            "estimate\n",
            "associations\n",
            "observed\n",
            "positive\n",
            "gradient\n",
            "incidence\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "nonsudden\n",
            "cardiac\n",
            "death\n",
            "nscd\n",
            "nonfatal\n",
            "mi\n",
            "association\n",
            "decreasing\n",
            "levels\n",
            "albumin\n",
            "death\n",
            "ventricular\n",
            "arrhythmias\n",
            "metaanalysisaims\n",
            "risk\n",
            "stratification\n",
            "patients\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "remains\n",
            "challenge\n",
            "brain\n",
            "natriuretic\n",
            "peptide\n",
            "bnp\n",
            "predicts\n",
            "overall\n",
            "mortality\n",
            "heart\n",
            "screening\n",
            "international\n",
            "footballbackground\n",
            "competitive\n",
            "sport\n",
            "serve\n",
            "trigger\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "majority\n",
            "athletes\n",
            "die\n",
            "suddenly\n",
            "previously\n",
            "unsuspected\n",
            "structural\n",
            "heart\n",
            "disease\n",
            "medical\n",
            "free\n",
            "t4\n",
            "associated\n",
            "qtcinterval\n",
            "prolongation\n",
            "risk\n",
            "factor\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "hyperthyroidism\n",
            "associated\n",
            "scd\n",
            "case\n",
            "reports\n",
            "populationbased\n",
            "studies\n",
            "confirming\n",
            "aim\n",
            "investigate\n",
            "small\n",
            "proportion\n",
            "athletes\n",
            "however\n",
            "increased\n",
            "risk\n",
            "exerciserelated\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "result\n",
            "quiescent\n",
            "cardiac\n",
            "anomalies\n",
            "preventable\n",
            "nature\n",
            "deaths\n",
            "prompted\n",
            "medical\n",
            "sporting\n",
            "cardiac\n",
            "death\n",
            "young\n",
            "athletes\n",
            "international\n",
            "perspectivethe\n",
            "incidence\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "among\n",
            "young\n",
            "athletes\n",
            "estimated\n",
            "13\n",
            "per\n",
            "100000\n",
            "person\n",
            "years\n",
            "may\n",
            "\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "chinapreliminary\n",
            "data\n",
            "suggest\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "occur\n",
            "rate\n",
            "418100000\n",
            "population\n",
            "china\n",
            "accounting\n",
            "544000\n",
            "deaths\n",
            "annually\n",
            "70\n",
            "icd\n",
            "treatment\n",
            "primary\n",
            "prevention\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "heart\n",
            "failure\n",
            "trial\n",
            "$\n",
            "heft\n",
            "placebo\n",
            "arm\n",
            "predicted\n",
            "4year\n",
            "mortality\n",
            "grouped\n",
            "5\n",
            "equalsized\n",
            "risk\n",
            "primary\n",
            "ventricular\n",
            "fibrillation\n",
            "percutaneous\n",
            "coronary\n",
            "intervention\n",
            "erabackground\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "constitutes\n",
            "one\n",
            "prevalent\n",
            "modes\n",
            "death\n",
            "mainly\n",
            "caused\n",
            "primary\n",
            "ventricular\n",
            "fibrillation\n",
            "enrolled\n",
            "international\n",
            "lqts\n",
            "registry\n",
            "involved\n",
            "20\n",
            "patients\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "16\n",
            "patients\n",
            "aborted\n",
            "cardiac\n",
            "arrest\n",
            "aca\n",
            "34\n",
            "patients\n",
            "syncope\n",
            "3253\n",
            "life\n",
            "studyaims\n",
            "determine\n",
            "whether\n",
            "qrs\n",
            "duration\n",
            "predicts\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "patients\n",
            "left\n",
            "ventricular\n",
            "hypertrophy\n",
            "treated\n",
            "hypertension\n",
            "methods\n",
            "results\n",
            "48\n",
            "09\n",
            "il18\n",
            "c\n",
            "allele\n",
            "polymorphism\n",
            "associated\n",
            "lower\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "examined\n",
            "process\n",
            "polymorphism\n",
            "alters\n",
            "risk\n",
            "scd\n",
            "coronary\n",
            "artery\n",
            "disease\n",
            "cad\n",
            "carriersaims\n",
            "investigate\n",
            "outcome\n",
            "cardiac\n",
            "evaluation\n",
            "risk\n",
            "stratification\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "asymptomatic\n",
            "hypertrophic\n",
            "cardiomyopathy\n",
            "hcm\n",
            "mutation\n",
            "carriers\n",
            "methods\n",
            "results\n",
            "seventysix\n",
            "hcm\n",
            "role\n",
            "wave\n",
            "alternansthe\n",
            "proportion\n",
            "cardiovascular\n",
            "deaths\n",
            "attributable\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "rise\n",
            "herein\n",
            "lies\n",
            "rationale\n",
            "developing\n",
            "risk\n",
            "stratification\n",
            "strategies\n",
            "predict\n",
            "benefit\n",
            "protein\n",
            "nos1ap\n",
            "locus\n",
            "strongly\n",
            "related\n",
            "qt\n",
            "interval\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "risk\n",
            "factor\n",
            "recent\n",
            "report\n",
            "describes\n",
            "common\n",
            "variation\n",
            "nos1ap\n",
            "associated\n",
            "scd\n",
            "april\n",
            "2007\n",
            "patients\n",
            "classified\n",
            "meeting\n",
            "meeting\n",
            "enrollment\n",
            "criteria\n",
            "either\n",
            "maditii\n",
            "$\n",
            "heft\n",
            "trials\n",
            "multivariable\n",
            "analyses\n",
            "assessed\n",
            "association\n",
            "gender\n",
            "concordance\n",
            "trial\n",
            "criteria\n",
            "alternans\n",
            "investigated\n",
            "canine\n",
            "postmyocardial\n",
            "infarction\n",
            "mi\n",
            "model\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "methods\n",
            "results\n",
            "postmi\n",
            "dogs\n",
            "preserved\n",
            "left\n",
            "ventricular\n",
            "lv\n",
            "function\n",
            "susceptibility\n",
            "coronary\n",
            "artery\n",
            "anomalies\n",
            "ccaas\n",
            "rare\n",
            "welldescribed\n",
            "cause\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "anomalous\n",
            "origin\n",
            "coronary\n",
            "artery\n",
            "opposite\n",
            "sinus\n",
            "frequently\n",
            "associated\n",
            "myocardial\n",
            "ischemia\n",
            "dilated\n",
            "cardiomyopathiesdilated\n",
            "cardiomyopathy\n",
            "dcm\n",
            "prevalent\n",
            "form\n",
            "cardiomyopathy\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "remains\n",
            "common\n",
            "event\n",
            "young\n",
            "patients\n",
            "disorder\n",
            "highly\n",
            "heterogeneous\n",
            "pediatric\n",
            "dcm\n",
            "literature\n",
            "supports\n",
            "fact\n",
            "many\n",
            "pediatric\n",
            "patients\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "exhibit\n",
            "warning\n",
            "signs\n",
            "symptoms\n",
            "use\n",
            "comprehensive\n",
            "cardiovascular\n",
            "screening\n",
            "questionnaires\n",
            "physical\n",
            "death\n",
            "predictive\n",
            "pediatric\n",
            "hcm\n",
            "background\n",
            "cardiac\n",
            "death\n",
            "hcm\n",
            "occurs\n",
            "suddenly\n",
            "$\n",
            "may\n",
            "nonsudden\n",
            "nonscd\n",
            "little\n",
            "data\n",
            "exists\n",
            "nonscd\n",
            "children\n",
            "risk\n",
            "factors\n",
            "ventricular\n",
            "fibrillationventricular\n",
            "tachycardia\n",
            "ventricular\n",
            "fibrillation\n",
            "important\n",
            "causes\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "particularly\n",
            "structural\n",
            "heart\n",
            "disease\n",
            "reduced\n",
            "left\n",
            "ventricular\n",
            "function\n",
            "important\n",
            "distinguish\n",
            "silent\n",
            "ischemia\n",
            "type\n",
            "ii\n",
            "swissi\n",
            "iithe\n",
            "occurrence\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "patients\n",
            "silent\n",
            "ischemia\n",
            "myocardial\n",
            "infarction\n",
            "mi\n",
            "factors\n",
            "facilitating\n",
            "scd\n",
            "unknown\n",
            "ablation\n",
            "obstructive\n",
            "hypertrophic\n",
            "cardiomyopathywe\n",
            "sought\n",
            "assess\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "ventricular\n",
            "arrhythmia\n",
            "alcohol\n",
            "septal\n",
            "ablation\n",
            "asa\n",
            "obstructive\n",
            "hypertrophic\n",
            "cardiomyopathy\n",
            "asa\n",
            "cause\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "young\n",
            "peoplebackground\n",
            "europe\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "one\n",
            "common\n",
            "causes\n",
            "death\n",
            "although\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "usually\n",
            "happens\n",
            "prevention\n",
            "treatmentthe\n",
            "incidence\n",
            "fatal\n",
            "nonfatal\n",
            "cardiovascular\n",
            "disease\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "myocardial\n",
            "infarction\n",
            "others\n",
            "varies\n",
            "depending\n",
            "conventional\n",
            "risk\n",
            "factors\n",
            "however\n",
            "western\n",
            "proper\n",
            "treatment\n",
            "followed\n",
            "observe\n",
            "allcause\n",
            "death\n",
            "rate\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "rate\n",
            "cox\n",
            "proportionalhazards\n",
            "regression\n",
            "model\n",
            "used\n",
            "investigate\n",
            "clinical\n",
            "factors\n",
            "affect\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "hcm\n",
            "$\n",
            "icd\n",
            "alphabet\n",
            "risks\n",
            "gaps\n",
            "patient\n",
            "counseling\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "cmr\n",
            "parameters\n",
            "spontaneous\n",
            "ventricular\n",
            "tachycardia\n",
            "vt\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "hypertrophic\n",
            "cardiomyopathy\n",
            "hcm\n",
            "patients\n",
            "methods\n",
            "results\n",
            "one\n",
            "hundred\n",
            "eight\n",
            "consecutive\n",
            "probtype\n",
            "natriuretic\n",
            "peptide\n",
            "ntprobnp\n",
            "found\n",
            "predict\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "patients\n",
            "known\n",
            "cardiac\n",
            "disease\n",
            "creactive\n",
            "protein\n",
            "levels\n",
            "found\n",
            "predict\n",
            "risk\n",
            "extremely\n",
            "high\n",
            "annual\n",
            "death\n",
            "rate\n",
            "approximately\n",
            "23\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "single\n",
            "largest\n",
            "cause\n",
            "death\n",
            "dialysis\n",
            "patients\n",
            "accounting\n",
            "approximately\n",
            "60\n",
            "cardiac\n",
            "study\n",
            "elucidate\n",
            "genetic\n",
            "markers\n",
            "associated\n",
            "elevated\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "collected\n",
            "autopsy\n",
            "material\n",
            "1999\n",
            "2001\n",
            "process\n",
            "forensic\n",
            "medical\n",
            "pathologoanatomical\n",
            "examination\n",
            "patients\n",
            "succumb\n",
            "either\n",
            "progressive\n",
            "congestive\n",
            "heart\n",
            "failure\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "implantable\n",
            "cardioverter\n",
            "defibrillator\n",
            "icd\n",
            "offers\n",
            "near\n",
            "sure\n",
            "means\n",
            "preventing\n",
            "scd\n",
            "united\n",
            "kingdom\n",
            "pathological\n",
            "studyobjective\n",
            "characterise\n",
            "demographics\n",
            "aetiology\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "athletes\n",
            "referred\n",
            "tertiary\n",
            "cardiac\n",
            "pathology\n",
            "centre\n",
            "uk\n",
            "design\n",
            "retrospective\n",
            "noncase\n",
            "perceptions\n",
            "policies\n",
            "economicfinancial\n",
            "impactthis\n",
            "study\n",
            "based\n",
            "consideration\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "incidence\n",
            "increasing\n",
            "todays\n",
            "society\n",
            "despite\n",
            "efforts\n",
            "clinicians\n",
            "great\n",
            "lack\n",
            "awareness\n",
            "results\n",
            "compared\n",
            "total\n",
            "mortality\n",
            "total\n",
            "cardiovascular\n",
            "mortality\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "spontaneous\n",
            "episodes\n",
            "ventricular\n",
            "arrhythmia\n",
            "sustained\n",
            "ventricular\n",
            "tachycardia\n",
            "vt\n",
            "ventricular\n",
            "fibrillation\n",
            "furthermore\n",
            "presence\n",
            "sths\n",
            "strong\n",
            "independent\n",
            "predictor\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "latter\n",
            "occurred\n",
            "eight\n",
            "patients\n",
            "sign\n",
            "842\n",
            "19\n",
            "none\n",
            "patients\n",
            "without\n",
            "approximately\n",
            "1\n",
            "500\n",
            "population\n",
            "serious\n",
            "complication\n",
            "hcm\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "first\n",
            "manifestation\n",
            "disease\n",
            "however\n",
            "forms\n",
            "benign\n",
            "prognosis\n",
            "jeopardize\n",
            "patients\n",
            "health\n",
            "\n",
            "\n",
            "\n",
            "ventricular\n",
            "arrhythmias\n",
            "sudden\n",
            "cardiac\n",
            "deathsudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "natural\n",
            "death\n",
            "caused\n",
            "cardiac\n",
            "reasons\n",
            "characterized\n",
            "sudden\n",
            "loss\n",
            "consciousness\n",
            "within\n",
            "therapies\n",
            "provided\n",
            "range\n",
            "management\n",
            "options\n",
            "patients\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "since\n",
            "interventions\n",
            "come\n",
            "attendant\n",
            "risks\n",
            "however\n",
            "stratification\n",
            "procedures\n",
            "aimed\n",
            "identifying\n",
            "death\n",
            "metaanalysis\n",
            "randomized\n",
            "controlled\n",
            "trialsaims\n",
            "patients\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "eligible\n",
            "access\n",
            "implantable\n",
            "cardioverter\n",
            "defibrillator\n",
            "icd\n",
            "implantation\n",
            "conflicting\n",
            "data\n",
            "regarding\n",
            "lowrisk\n",
            "population\n",
            "enhanced\n",
            "specificity\n",
            "heightened\n",
            "twave\n",
            "alternansaims\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "due\n",
            "cardiac\n",
            "electrical\n",
            "instability\n",
            "postulated\n",
            "prediction\n",
            "mode\n",
            "death\n",
            "exercise\n",
            "capacity\n",
            "results\n",
            "nurses\n",
            "health\n",
            "studyobjectives\n",
            "assessed\n",
            "association\n",
            "depression\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "cardiac\n",
            "events\n",
            "among\n",
            "individuals\n",
            "without\n",
            "baseline\n",
            "coronary\n",
            "heart\n",
            "disease\n",
            "chd\n",
            "testing\n",
            "clinical\n",
            "practicethe\n",
            "public\n",
            "health\n",
            "challenge\n",
            "preventing\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "significant\n",
            "various\n",
            "risk\n",
            "factors\n",
            "scd\n",
            "well\n",
            "articulated\n",
            "clinical\n",
            "trials\n",
            "importance\n",
            "factors\n",
            "sudden\n",
            "cardiac\n",
            "deathidentifying\n",
            "risk\n",
            "factors\n",
            "predict\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "important\n",
            "area\n",
            "investigation\n",
            "current\n",
            "techniques\n",
            "limited\n",
            "several\n",
            "issues\n",
            "result\n",
            "limitations\n",
            "vt\n",
            "induction\n",
            "testsyear\n",
            "conclusions\n",
            "current\n",
            "icd\n",
            "therapy\n",
            "guidelines\n",
            "primary\n",
            "prevention\n",
            "$\n",
            "postmi\n",
            "demand\n",
            "substantial\n",
            "increase\n",
            "service\n",
            "provision\n",
            "worldwide\n",
            "\n",
            "\n",
            "\n",
            "64169\n",
            "presence\n",
            "sustained\n",
            "vt\n",
            "associated\n",
            "high\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "lowest\n",
            "12month\n",
            "overall\n",
            "survival\n",
            "p\n",
            "lt\n",
            "00001\n",
            "conclusion\n",
            "incidence\n",
            "sustained\n",
            "clinical\n",
            "application\n",
            "implantable\n",
            "cardiovertersdefibrillators\n",
            "icd\n",
            "method\n",
            "prevention\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "work\n",
            "definition\n",
            "epidemiology\n",
            "mechanism\n",
            "nosological\n",
            "causes\n",
            "scd\n",
            "presented\n",
            "followed\n",
            "historical\n",
            "cardiovascular\n",
            "cv\n",
            "death\n",
            "occurred\n",
            "39\n",
            "patients\n",
            "28\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "cv\n",
            "death\n",
            "scd\n",
            "lvef\n",
            "lt30\n",
            "ventricular\n",
            "premature\n",
            "beatsh\n",
            "gt20\n",
            "independent\n",
            "communitybased\n",
            "populationsbackground\n",
            "ecg\n",
            "qt\n",
            "interval\n",
            "associated\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "previous\n",
            "genomewide\n",
            "association\n",
            "study\n",
            "demonstrated\n",
            "allelic\n",
            "variants\n",
            "rs10494366\n",
            "rs4657139\n",
            "nitric\n",
            "channel\n",
            "genes\n",
            "telomere\n",
            "shortening\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "victimssudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "caused\n",
            "number\n",
            "reasons\n",
            "previous\n",
            "works\n",
            "identified\n",
            "genetic\n",
            "causes\n",
            "alterations\n",
            "dna\n",
            "2year\n",
            "followup\n",
            "causespecific\n",
            "mortality\n",
            "rates\n",
            "predictors\n",
            "implantation\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "progressive\n",
            "heart\n",
            "failure\n",
            "hf\n",
            "death\n",
            "assessed\n",
            "using\n",
            "competing\n",
            "risk\n",
            "methodology\n",
            "populationextremes\n",
            "electrocardiographic\n",
            "qt\n",
            "interval\n",
            "associated\n",
            "increased\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "thus\n",
            "identification\n",
            "characterization\n",
            "genetic\n",
            "variants\n",
            "modulate\n",
            "qt\n",
            "interval\n",
            "may\n",
            "elucidate\n",
            "predictive\n",
            "value\n",
            "cardiac\n",
            "iodine123\n",
            "metaiodobenzylguanidine\n",
            "mibg\n",
            "imaging\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "signalaveraged\n",
            "electrocardiogram\n",
            "saecg\n",
            "heart\n",
            "rate\n",
            "variability\n",
            "hrv\n",
            "qt\n",
            "dispersion\n",
            "patients\n",
            "parameters\n",
            "capacity\n",
            "evaluate\n",
            "risk\n",
            "total\n",
            "cardiovascular\n",
            "mortality\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "methods\n",
            "results\n",
            "finnish\n",
            "cardiovascular\n",
            "study\n",
            "fincavas\n",
            "enrolled\n",
            "consecutive\n",
            "patients\n",
            "n\n",
            "qt\n",
            "interval\n",
            "qtc\n",
            "associated\n",
            "independent\n",
            "increased\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "evaluated\n",
            "determinants\n",
            "prolonged\n",
            "qtc\n",
            "relationship\n",
            "prolonged\n",
            "qtc\n",
            "scd\n",
            "risk\n",
            "among\n",
            "ventricular\n",
            "tachyarrhythmic\n",
            "events\n",
            "vtes\n",
            "patients\n",
            "risk\n",
            "factors\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "without\n",
            "prior\n",
            "history\n",
            "sustained\n",
            "vtes\n",
            "primary\n",
            "prevention\n",
            "patients\n",
            "recent\n",
            "studies\n",
            "patients\n",
            "risk\n",
            "sudden\n",
            "cardiac\n",
            "deaththe\n",
            "mechanisms\n",
            "underlying\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "complex\n",
            "diverse\n",
            "therefore\n",
            "correct\n",
            "application\n",
            "marker\n",
            "risk\n",
            "stratify\n",
            "patients\n",
            "appropriate\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "prevention\n",
            "lessons\n",
            "past\n",
            "strategies\n",
            "futuresudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "commonly\n",
            "caused\n",
            "ventricular\n",
            "fibrillation\n",
            "vf\n",
            "single\n",
            "largest\n",
            "cohort\n",
            "victims\n",
            "population\n",
            "review\n",
            "trials\n",
            "concerning\n",
            "effects\n",
            "omega3\n",
            "fatty\n",
            "acids\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "cardiac\n",
            "death\n",
            "allcause\n",
            "mortality\n",
            "coronary\n",
            "heart\n",
            "disease\n",
            "chd\n",
            "patients\n",
            "methods\n",
            "\n",
            "\n",
            "advances\n",
            "prevention\n",
            "sudden\n",
            "cardiac\n",
            "death\n",
            "youngsudden\n",
            "cardiac\n",
            "death\n",
            "$\n",
            "tragic\n",
            "devastating\n",
            "complication\n",
            "number\n",
            "cardiovascular\n",
            "diseases\n",
            "although\n",
            "coronary\n",
            "artery\n",
            "disease\n",
            "increasing\n",
            "immigrant\n",
            "population\n",
            "republic\n",
            "ireland\n",
            "number\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "seen\n",
            "paediatric\n",
            "hospitals\n",
            "climbing\n",
            "case\n",
            "report\n",
            "review\n",
            "anaesthetic\n",
            "implications\n",
            "outcome\n",
            "sickle\n",
            "cell\n",
            "diseasebackground\n",
            "prevalence\n",
            "iron\n",
            "deficiency\n",
            "pediatric\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "difficult\n",
            "describe\n",
            "standard\n",
            "markers\n",
            "iron\n",
            "metabolism\n",
            "altered\n",
            "inflammatory\n",
            "state\n",
            "soluble\n",
            "blood\n",
            "cultures\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "diseasebackground\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "considered\n",
            "risk\n",
            "bacteremia\n",
            "especially\n",
            "present\n",
            "emergency\n",
            "department\n",
            "ed\n",
            "fever\n",
            "objective\n",
            "animals\n",
            "however\n",
            "implications\n",
            "potent\n",
            "components\n",
            "bmp6\n",
            "smad1\n",
            "smad2\n",
            "apt\n",
            "pathway\n",
            "$\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "pathology\n",
            "orthopedic\n",
            "complications\n",
            "orthoss\n",
            "poorly\n",
            "elucidated\n",
            "substantially\n",
            "progressive\n",
            "vascular\n",
            "disease\n",
            "results\n",
            "high\n",
            "mortality\n",
            "morbidity\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "patients\n",
            "pah\n",
            "diagnosis\n",
            "invasive\n",
            "via\n",
            "right\n",
            "heart\n",
            "catheterization\n",
            "manual\n",
            "measurements\n",
            "questionnaires\n",
            "40\n",
            "taped\n",
            "interviews\n",
            "young\n",
            "people\n",
            "living\n",
            "sickle\n",
            "cell\n",
            "disorder\n",
            "$\n",
            "england\n",
            "found\n",
            "students\n",
            "scd\n",
            "faced\n",
            "dilemma\n",
            "whether\n",
            "disclose\n",
            "sickle\n",
            "cell\n",
            "multidisciplinary\n",
            "team\n",
            "charged\n",
            "improving\n",
            "analgesic\n",
            "management\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "team\n",
            "developed\n",
            "nurseinitiated\n",
            "analgesic\n",
            "protocol\n",
            "scd\n",
            "patients\n",
            "implemented\n",
            "baseline\n",
            "data\n",
            "disease\n",
            "report\n",
            "international\n",
            "symposium\n",
            "workshop\n",
            "cotonou\n",
            "republic\n",
            "beninsickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "one\n",
            "common\n",
            "monogenetic\n",
            "diseases\n",
            "worldwide\n",
            "although\n",
            "advances\n",
            "management\n",
            "scd\n",
            "highincome\n",
            "sets\n",
            "important\n",
            "source\n",
            "data\n",
            "health\n",
            "services\n",
            "research\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "thalassemia\n",
            "paper\n",
            "identifies\n",
            "describes\n",
            "major\n",
            "us\n",
            "healthcare\n",
            "administrative\n",
            "databases\n",
            "use\n",
            "pediatric\n",
            "medical\n",
            "expenditures\n",
            "usbackground\n",
            "although\n",
            "known\n",
            "people\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "relatively\n",
            "high\n",
            "utilization\n",
            "medical\n",
            "care\n",
            "previous\n",
            "estimates\n",
            "scdattributable\n",
            "expenditures\n",
            "limited\n",
            "cell\n",
            "disease\n",
            "pregnancy\n",
            "outcomes\n",
            "women\n",
            "african\n",
            "descentbackground\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "severe\n",
            "hematologic\n",
            "condition\n",
            "presents\n",
            "unique\n",
            "complications\n",
            "among\n",
            "affected\n",
            "pregnant\n",
            "women\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "descriptive\n",
            "study\n",
            "19992007background\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "often\n",
            "use\n",
            "emergency\n",
            "department\n",
            "services\n",
            "obtain\n",
            "medical\n",
            "care\n",
            "limited\n",
            "information\n",
            "health\n",
            "status\n",
            "use\n",
            "health\n",
            "services\n",
            "among\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "purpose\n",
            "study\n",
            "provides\n",
            "estimates\n",
            "cooccurring\n",
            "conditions\n",
            "health\n",
            "impact\n",
            "utilization\n",
            "barriers\n",
            "estimates\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "usbackground\n",
            "number\n",
            "individuals\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "us\n",
            "unknown\n",
            "determination\n",
            "burden\n",
            "disease\n",
            "healthcare\n",
            "issues\n",
            "policies\n",
            "best\n",
            "served\n",
            "plgf\n",
            "novel\n",
            "inducer\n",
            "plasminogen\n",
            "activator\n",
            "inhibitor1\n",
            "pai1\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "scd\n",
            "characterized\n",
            "prothrombotic\n",
            "state\n",
            "plasminogen\n",
            "activator\n",
            "inhibitor1\n",
            "influence\n",
            "pregnancy\n",
            "bone\n",
            "mass\n",
            "sickle\n",
            "cell\n",
            "anemiabackground\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "causes\n",
            "osteopenia\n",
            "osteoporosis\n",
            "osteopenia\n",
            "may\n",
            "influenced\n",
            "pregnancy\n",
            "objective\n",
            "find\n",
            "effect\n",
            "severity\n",
            "vascular\n",
            "dysfunction\n",
            "pulmonary\n",
            "hypertensionpulmonary\n",
            "hypertension\n",
            "ph\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "emerging\n",
            "important\n",
            "clinical\n",
            "problem\n",
            "singleinstitution\n",
            "adult\n",
            "cohort\n",
            "365\n",
            "patients\n",
            "investigated\n",
            "relevance\n",
            "sickle\n",
            "cell\n",
            "diseasebackground\n",
            "objective\n",
            "clinical\n",
            "hallmarks\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "vasoocclusive\n",
            "crises\n",
            "voc\n",
            "triggered\n",
            "red\n",
            "blood\n",
            "cells\n",
            "rbc\n",
            "stiffening\n",
            "abnormal\n",
            "examined\n",
            "factors\n",
            "influencing\n",
            "research\n",
            "participation\n",
            "among\n",
            "individuals\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "caregivers\n",
            "children\n",
            "scd\n",
            "procedure\n",
            "distributed\n",
            "32item\n",
            "selfadministered\n",
            "survey\n",
            "parents\n",
            "legal\n",
            "hemoglobinopathiesthe\n",
            "impact\n",
            "hemoglobinopathies\n",
            "healthcare\n",
            "united\n",
            "states\n",
            "particularly\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "significant\n",
            "enactment\n",
            "sickle\n",
            "cell\n",
            "anemia\n",
            "control\n",
            "act\n",
            "1972\n",
            "significantly\n",
            "increased\n",
            "\n",
            "\n",
            "\n",
            "ocular\n",
            "manifestations\n",
            "sickle\n",
            "cell\n",
            "diseasesickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "common\n",
            "genetic\n",
            "disease\n",
            "worldwide\n",
            "increase\n",
            "life\n",
            "expectancy\n",
            "scd\n",
            "patients\n",
            "recent\n",
            "adolescents\n",
            "sickle\n",
            "cell\n",
            "diseasethe\n",
            "survival\n",
            "young\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "improved\n",
            "less\n",
            "known\n",
            "older\n",
            "children\n",
            "adolescents\n",
            "studied\n",
            "dallas\n",
            "newborn\n",
            "cohort\n",
            "study\n",
            "evaluate\n",
            "integrity\n",
            "peripheral\n",
            "central\n",
            "auditory\n",
            "systems\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "patients\n",
            "electrophysiological\n",
            "evaluation\n",
            "utilizing\n",
            "auditory\n",
            "evoked\n",
            "potentials\n",
            "comparing\n",
            "results\n",
            "obtained\n",
            "bosentan\n",
            "results\n",
            "asset\n",
            "studiesdopplerdefined\n",
            "pulmonary\n",
            "hypertension\n",
            "ph\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "associated\n",
            "40\n",
            "mortality\n",
            "40\n",
            "months\n",
            "assess\n",
            "effect\n",
            "bosentan\n",
            "scdph\n",
            "two\n",
            "difference\n",
            "time\n",
            "initial\n",
            "analgesic\n",
            "patients\n",
            "acute\n",
            "pain\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "versus\n",
            "renal\n",
            "colic\n",
            "rc\n",
            "identify\n",
            "factors\n",
            "contributing\n",
            "variance\n",
            "time\n",
            "analgesic\n",
            "cell\n",
            "diseasethis\n",
            "study\n",
            "investigated\n",
            "hydroxyurea\n",
            "use\n",
            "people\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "outside\n",
            "research\n",
            "setting\n",
            "pharmacy\n",
            "data\n",
            "outpatient\n",
            "visits\n",
            "hospital\n",
            "admissions\n",
            "length\n",
            "cell\n",
            "diseasedespite\n",
            "frequent\n",
            "episodes\n",
            "severe\n",
            "recurrent\n",
            "pain\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "sensory\n",
            "pain\n",
            "outpatient\n",
            "adults\n",
            "scd\n",
            "lacks\n",
            "sufficient\n",
            "characterization\n",
            "furthermore\n",
            "pivotal\n",
            "phosphodiesterase\n",
            "3a\n",
            "activitywhilst\n",
            "high\n",
            "procoagulant\n",
            "activity\n",
            "reported\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "precise\n",
            "role\n",
            "platelets\n",
            "plts\n",
            "scd\n",
            "inflammatory\n",
            "vasoocclusive\n",
            "processes\n",
            "unclear\n",
            "adhesion\n",
            "initial\n",
            "oxygen\n",
            "saturation\n",
            "measurementa\n",
            "prospective\n",
            "cohort\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "evaluated\n",
            "determine\n",
            "variability\n",
            "daytime\n",
            "pulse\n",
            "oximetry\n",
            "among\n",
            "three\n",
            "measurements\n",
            "approximately\n",
            "vasoocclusive\n",
            "crisisthe\n",
            "pathogenesis\n",
            "sickle\n",
            "vasoocclusive\n",
            "crisis\n",
            "voc\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "patients\n",
            "involves\n",
            "accumulation\n",
            "rigid\n",
            "sickle\n",
            "cells\n",
            "stimulation\n",
            "ongoing\n",
            "inflammatory\n",
            "response\n",
            "blood\n",
            "flow\n",
            "velocity\n",
            "specific\n",
            "language\n",
            "abilities\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "thirtynine\n",
            "children\n",
            "ages\n",
            "5\n",
            "8\n",
            "years\n",
            "old\n",
            "highrisk\n",
            "genotypes\n",
            "scd\n",
            "disease\n",
            "haemoglobin\n",
            "electrophoresis\n",
            "survey\n",
            "students\n",
            "tertiary\n",
            "institutionbackground\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "commonest\n",
            "genetic\n",
            "disease\n",
            "worldwide\n",
            "sickle\n",
            "cell\n",
            "control\n",
            "strategies\n",
            "premarital\n",
            "genetic\n",
            "blood\n",
            "transfusions\n",
            "prevent\n",
            "treat\n",
            "serious\n",
            "complications\n",
            "related\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "inevitably\n",
            "lead\n",
            "iron\n",
            "overload\n",
            "complications\n",
            "terms\n",
            "morbidity\n",
            "mortality\n",
            "transfusion\n",
            "history\n",
            "therapy\n",
            "offers\n",
            "promise\n",
            "ameliorating\n",
            "clinical\n",
            "course\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "hydroxyurea\n",
            "prototypic\n",
            "therapeutic\n",
            "option\n",
            "administered\n",
            "minimal\n",
            "side\n",
            "effects\n",
            "relatively\n",
            "wide\n",
            "hospitalizations\n",
            "among\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "diseaseobjective\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "considered\n",
            "high\n",
            "risk\n",
            "complications\n",
            "influenza\n",
            "infection\n",
            "despite\n",
            "minimal\n",
            "published\n",
            "data\n",
            "costeffectiveness\n",
            "universal\n",
            "screening\n",
            "programme\n",
            "st\n",
            "maartenobjective\n",
            "estimate\n",
            "incidence\n",
            "sicklecell\n",
            "disease\n",
            "$\n",
            "aruba\n",
            "st\n",
            "maarten\n",
            "determine\n",
            "whether\n",
            "universal\n",
            "screening\n",
            "would\n",
            "costeffective\n",
            "according\n",
            "infection\n",
            "cytolysin\n",
            "toxicity\n",
            "mouse\n",
            "model\n",
            "sickle\n",
            "cell\n",
            "diseasesickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "characterized\n",
            "intravascular\n",
            "hemolysis\n",
            "inflammation\n",
            "coupled\n",
            "400fold\n",
            "greater\n",
            "incidence\n",
            "invasive\n",
            "pneumococcal\n",
            "acute\n",
            "chest\n",
            "syndrome\n",
            "people\n",
            "sickle\n",
            "cell\n",
            "diseasebackground\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "inherited\n",
            "autosomal\n",
            "recessive\n",
            "blood\n",
            "condition\n",
            "one\n",
            "prevalent\n",
            "genetic\n",
            "blood\n",
            "diseases\n",
            "\n",
            "\n",
            "glomerular\n",
            "lesions\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "diseasesickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "increasing\n",
            "cause\n",
            "chronic\n",
            "kidney\n",
            "disease\n",
            "spectrum\n",
            "glomerular\n",
            "lesions\n",
            "underlying\n",
            "mechanisms\n",
            "red\n",
            "cell\n",
            "aplasia\n",
            "trca\n",
            "acute\n",
            "anaemia\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "report\n",
            "three\n",
            "cases\n",
            "patients\n",
            "developed\n",
            "nephrotic\n",
            "syndrome\n",
            "ns\n",
            "chronic\n",
            "sequelae\n",
            "estimates\n",
            "inpatient\n",
            "use\n",
            "laxatives\n",
            "opioid\n",
            "administration\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "identify\n",
            "associations\n",
            "individual\n",
            "hospitallevel\n",
            "factors\n",
            "laxative\n",
            "prescription\n",
            "procedure\n",
            "analyzed\n",
            "data\n",
            "growth\n",
            "factor\n",
            "plgf\n",
            "released\n",
            "immature\n",
            "erythrocytes\n",
            "elevated\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "previous\n",
            "data\n",
            "generated\n",
            "vitro\n",
            "suggest\n",
            "plgf\n",
            "may\n",
            "play\n",
            "role\n",
            "pathophysiology\n",
            "leading\n",
            "premature\n",
            "deaths\n",
            "adult\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "diseasesickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "associated\n",
            "early\n",
            "mortality\n",
            "sought\n",
            "determine\n",
            "incidence\n",
            "cause\n",
            "risk\n",
            "factors\n",
            "death\n",
            "shunts\n",
            "including\n",
            "patent\n",
            "foramen\n",
            "ovale\n",
            "pfo\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "stroke\n",
            "study\n",
            "design\n",
            "performed\n",
            "transthoracic\n",
            "echocardiogram\n",
            "tte\n",
            "40\n",
            "children\n",
            "scd\n",
            "producing\n",
            "antibodies\n",
            "highincidence\n",
            "rh\n",
            "antigens\n",
            "particularly\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "advances\n",
            "use\n",
            "hydroxyureaclinical\n",
            "experience\n",
            "hydroxyurea\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "accumulating\n",
            "past\n",
            "25\n",
            "years\n",
            "bulk\n",
            "current\n",
            "evidence\n",
            "suggests\n",
            "hydroxyurea\n",
            "welltolerated\n",
            "therapies\n",
            "sickle\n",
            "cell\n",
            "diseasedespite\n",
            "increased\n",
            "understanding\n",
            "pathophysiology\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "remains\n",
            "paucity\n",
            "available\n",
            "agents\n",
            "prevention\n",
            "treatment\n",
            "specific\n",
            "scdrelated\n",
            "complications\n",
            "recently\n",
            "part\n",
            "processa\n",
            "physician\n",
            "diagnosis\n",
            "asthma\n",
            "children\n",
            "adults\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "associated\n",
            "increased\n",
            "rates\n",
            "pain\n",
            "acute\n",
            "chest\n",
            "syndrome\n",
            "acs\n",
            "episodes\n",
            "premature\n",
            "ameliorates\n",
            "organ\n",
            "oxidative\n",
            "stress\n",
            "transgenicknockout\n",
            "sickle\n",
            "micein\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "events\n",
            "originating\n",
            "hemoglobin\n",
            "polymerization\n",
            "intravascular\n",
            "sickling\n",
            "lead\n",
            "reperfusion\n",
            "injury\n",
            "hemolysis\n",
            "\n",
            "morphometry\n",
            "thoracolumbar\n",
            "vertebrae\n",
            "sickle\n",
            "cell\n",
            "diseasepatients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "deformities\n",
            "vertebral\n",
            "fractures\n",
            "due\n",
            "osteoporosis\n",
            "may\n",
            "require\n",
            "surgery\n",
            "spinal\n",
            "surgeons\n",
            "erarecent\n",
            "investigations\n",
            "identified\n",
            "syndrome\n",
            "hemolysisassociated\n",
            "vasculopathy\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "features\n",
            "severe\n",
            "hemolytic\n",
            "anemia\n",
            "leads\n",
            "scavenging\n",
            "nitric\n",
            "oxide\n",
            "biochemical\n",
            "precursor\n",
            "pathophysiological\n",
            "consequences\n",
            "transfusional\n",
            "iron\n",
            "overload\n",
            "transfused\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "appear\n",
            "less\n",
            "likely\n",
            "develop\n",
            "extrahepatic\n",
            "complications\n",
            "possibly\n",
            "plasma\n",
            "nontransferrinbound\n",
            "iron\n",
            "pressing\n",
            "reason\n",
            "importance\n",
            "develop\n",
            "prevention\n",
            "program\n",
            "betathalassemia\n",
            "egypt\n",
            "sicklecell\n",
            "disease\n",
            "$\n",
            "frequent\n",
            "egypt\n",
            "except\n",
            "oases\n",
            "carrier\n",
            "rate\n",
            "varies\n",
            "9\n",
            "22\n",
            "objectives\n",
            "\n",
            "\n",
            "current\n",
            "understanding\n",
            "management\n",
            "sickle\n",
            "cell\n",
            "diseasesickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "commonest\n",
            "monogenetic\n",
            "disorder\n",
            "worldwide\n",
            "represents\n",
            "major\n",
            "public\n",
            "health\n",
            "burden\n",
            "significant\n",
            "states\n",
            "newborn\n",
            "screening\n",
            "programs\n",
            "universally\n",
            "identify\n",
            "newborns\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "heterozygote\n",
            "carriers\n",
            "sickle\n",
            "cell\n",
            "trait\n",
            "sct\n",
            "although\n",
            "consensus\n",
            "disclose\n",
            "sct\n",
            "attempts\n",
            "30\n",
            "male\n",
            "37\n",
            "female\n",
            "black\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "scd\n",
            "condition\n",
            "characterized\n",
            "chronic\n",
            "unpredictable\n",
            "pains\n",
            "psychosocial\n",
            "distress\n",
            "thirtysix\n",
            "percent\n",
            "sickle\n",
            "cell\n",
            "vasocclusive\n",
            "crisesseveral\n",
            "lines\n",
            "evidence\n",
            "suggest\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "associated\n",
            "chronic\n",
            "inflammatory\n",
            "state\n",
            "study\n",
            "70\n",
            "children\n",
            "scd\n",
            "steady\n",
            "state\n",
            "heme\n",
            "oxygenase\n",
            "activity\n",
            "effects\n",
            "vascular\n",
            "adhesion\n",
            "reactivityobjective\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "characterized\n",
            "extensive\n",
            "hemolysis\n",
            "increased\n",
            "cellular\n",
            "adhesion\n",
            "vasoocclusion\n",
            "tissues\n",
            "sickle\n",
            "patients\n",
            "blood\n",
            "cell\n",
            "rbc\n",
            "transfusion\n",
            "well\n",
            "described\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "nonscd\n",
            "patients\n",
            "awareness\n",
            "condition\n",
            "important\n",
            "subsequent\n",
            "transfusion\n",
            "may\n",
            "exacerbate\n",
            "hemolysis\n",
            "f\n",
            "phenotypelow\n",
            "bone\n",
            "mineral\n",
            "density\n",
            "bmd\n",
            "common\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "patients\n",
            "ages\n",
            "due\n",
            "erythropoietic\n",
            "stress\n",
            "consequent\n",
            "bone\n",
            "marrow\n",
            "hyperplasia\n",
            "kuwaiti\n",
            "peripheral\n",
            "arterial\n",
            "tonometry\n",
            "assessing\n",
            "endothelial\n",
            "dysfunction\n",
            "pediatric\n",
            "sickle\n",
            "cell\n",
            "diseasebackground\n",
            "$\n",
            "characterized\n",
            "hemolysis\n",
            "oxidative\n",
            "stress\n",
            "resulting\n",
            "endothelial\n",
            "dysfunction\n",
            "edf\n",
            "peripheral\n",
            "arterial\n",
            "hemodilution\n",
            "anh\n",
            "rarely\n",
            "used\n",
            "perioperative\n",
            "management\n",
            "homozygous\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "patients\n",
            "undergoing\n",
            "surgery\n",
            "hereby\n",
            "present\n",
            "case\n",
            "anh\n",
            "used\n",
            "successfully\n",
            "22\n",
            "acute\n",
            "splenic\n",
            "sequestrations\n",
            "sss\n",
            "potentially\n",
            "fatal\n",
            "complications\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "total\n",
            "splenectomies\n",
            "young\n",
            "patients\n",
            "may\n",
            "predispose\n",
            "higher\n",
            "risk\n",
            "overwhelming\n",
            "infections\n",
            "model\n",
            "simulation\n",
            "study\n",
            "implications\n",
            "blood\n",
            "flow\n",
            "regulation\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "nitrite\n",
            "react\n",
            "parallel\n",
            "adult\n",
            "oxy\n",
            "deoxyhemoglobin\n",
            "hba\n",
            "resulting\n",
            "oxidative\n",
            "denitrosylation\n",
            "replacement\n",
            "conditioning\n",
            "regimen\n",
            "modalities\n",
            "infusion\n",
            "essential\n",
            "cure\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "bone\n",
            "marrowbmstem\n",
            "cell\n",
            "transplantation\n",
            "gene\n",
            "therapy\n",
            "must\n",
            "established\n",
            "prior\n",
            "clinical\n",
            "hemolytic\n",
            "anemia\n",
            "newborn\n",
            "screening\n",
            "early\n",
            "treatment\n",
            "death\n",
            "rate\n",
            "among\n",
            "children\n",
            "$\n",
            "declined\n",
            "addition\n",
            "variety\n",
            "treatments\n",
            "introduced\n",
            "help\n",
            "manage\n",
            "various\n",
            "manifestations\n",
            "disease\n",
            "cell\n",
            "membrane\n",
            "proteomehydroxyurea\n",
            "hu\n",
            "effective\n",
            "drug\n",
            "treatment\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "main\n",
            "clinical\n",
            "benefit\n",
            "hu\n",
            "thought\n",
            "derive\n",
            "capacity\n",
            "increase\n",
            "fetal\n",
            "hemoglobin\n",
            "stroke\n",
            "prevention\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "diseaseobjectives\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "chronic\n",
            "transfusion\n",
            "maintain\n",
            "haemoglobin\n",
            "hbs\n",
            "30\n",
            "markedly\n",
            "decreases\n",
            "risk\n",
            "first\n",
            "neonatal\n",
            "screening\n",
            "sickle\n",
            "cell\n",
            "diseasebackground\n",
            "screening\n",
            "programs\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "utilize\n",
            "isoelectric\n",
            "focusing\n",
            "ief\n",
            "high\n",
            "performance\n",
            "liquid\n",
            "chromatography\n",
            "hplc\n",
            "detect\n",
            "evaluate\n",
            "role\n",
            "endoscopic\n",
            "retrograde\n",
            "cholangiopancreatography\n",
            "ercp\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "methods\n",
            "two\n",
            "hundred\n",
            "twenty\n",
            "four\n",
            "scd\n",
            "patients\n",
            "cholestatic\n",
            "jaundice\n",
            "cj\n",
            "review\n",
            "highlights\n",
            "recent\n",
            "advances\n",
            "iron\n",
            "metabolism\n",
            "relevant\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "scd\n",
            "common\n",
            "hemoglobinopathy\n",
            "results\n",
            "chronic\n",
            "inflammation\n",
            "improved\n",
            "understanding\n",
            "iron\n",
            "metabolism\n",
            "evaluate\n",
            "longterm\n",
            "efficacy\n",
            "safety\n",
            "hydroxyurea\n",
            "hu\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "thirtyfour\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "anemia\n",
            "hemoglobin\n",
            "hbshbs\n",
            "131\n",
            "hbsbeta0thal\n",
            "165\n",
            "\n",
            "\n",
            "definitions\n",
            "phenotypic\n",
            "manifestations\n",
            "sickle\n",
            "cell\n",
            "diseasesickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "pleiotropic\n",
            "genetic\n",
            "disorder\n",
            "hemoglobin\n",
            "profound\n",
            "multiorgan\n",
            "effects\n",
            "low\n",
            "prevalence\n",
            "scd\n",
            "pediatric\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "patients\n",
            "adult\n",
            "providerschildren\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "surviving\n",
            "past\n",
            "childhood\n",
            "well\n",
            "adulthood\n",
            "concerns\n",
            "raised\n",
            "going\n",
            "care\n",
            "children\n",
            "red\n",
            "blood\n",
            "cell\n",
            "rbc\n",
            "antigens\n",
            "occurs\n",
            "commonly\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "patients\n",
            "alloimmunization\n",
            "demonstrate\n",
            "increased\n",
            "risk\n",
            "new\n",
            "alloantibody\n",
            "formation\n",
            "subsequent\n",
            "transfusion\n",
            "associated\n",
            "osteomyelitis\n",
            "osteonecrosisbackground\n",
            "septic\n",
            "arthritis\n",
            "known\n",
            "complication\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "children\n",
            "association\n",
            "osteomyelitis\n",
            "osteonecrosis\n",
            "described\n",
            "however\n",
            "unclear\n",
            "whether\n",
            "association\n",
            "applies\n",
            "cystic\n",
            "fibrosisillinois\n",
            "introduced\n",
            "mandatory\n",
            "newborn\n",
            "screening\n",
            "nbs\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "1989\n",
            "cystic\n",
            "fibrosis\n",
            "cf\n",
            "2008\n",
            "examined\n",
            "maternal\n",
            "understanding\n",
            "nbs\n",
            "scd\n",
            "sickle\n",
            "cell\n",
            "diseaseclinical\n",
            "experts\n",
            "expressed\n",
            "concern\n",
            "underutilization\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "therapies\n",
            "including\n",
            "hydroxyurea\n",
            "prophylactic\n",
            "antibiotics\n",
            "iron\n",
            "chelation\n",
            "bone\n",
            "marrow\n",
            "transplantation\n",
            "qol\n",
            "measurements\n",
            "becoming\n",
            "vital\n",
            "part\n",
            "health\n",
            "outcome\n",
            "appraisal\n",
            "sicklecell\n",
            "disease\n",
            "$\n",
            "places\n",
            "substantial\n",
            "burden\n",
            "afflicted\n",
            "qol\n",
            "measure\n",
            "validated\n",
            "use\n",
            "jamaica\n",
            "sf36\n",
            "cell\n",
            "diseasebackground\n",
            "usage\n",
            "contraceptive\n",
            "methods\n",
            "young\n",
            "women\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "factors\n",
            "affecting\n",
            "usage\n",
            "unknown\n",
            "study\n",
            "design\n",
            "young\n",
            "women\n",
            "n132\n",
            "scd\n",
            "pulmonary\n",
            "hypertension\n",
            "phtn\n",
            "increasingly\n",
            "recognized\n",
            "serious\n",
            "complication\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "objective\n",
            "determine\n",
            "prevalence\n",
            "phtn\n",
            "identify\n",
            "factors\n",
            "associated\n",
            "phtn\n",
            "among\n",
            "children\n",
            "sibling\n",
            "bone\n",
            "marrow\n",
            "transplantation\n",
            "bmt\n",
            "children\n",
            "severe\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "1991\n",
            "2000\n",
            "determine\n",
            "children\n",
            "protected\n",
            "complications\n",
            "scd\n",
            "successful\n",
            "bmt\n",
            "extended\n",
            "patients\n",
            "ugandabackground\n",
            "blood\n",
            "transfusion\n",
            "integral\n",
            "part\n",
            "management\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "patients\n",
            "alloimmunization\n",
            "recognized\n",
            "complication\n",
            "red\n",
            "blood\n",
            "cell\n",
            "rbc\n",
            "transfusions\n",
            "consequences\n",
            "sickle\n",
            "cell\n",
            "diseasepathophysiology\n",
            "therapyover\n",
            "70000\n",
            "people\n",
            "live\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "united\n",
            "states\n",
            "multitudes\n",
            "worldwide\n",
            "2000\n",
            "afflicted\n",
            "babies\n",
            "born\n",
            "country\n",
            "year\n",
            "levels\n",
            "sickle\n",
            "cell\n",
            "diseaseinflammation\n",
            "angiogenesis\n",
            "importance\n",
            "pathophysiology\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "recently\n",
            "chemokine\n",
            "stromalderived\n",
            "factor1\n",
            "sdf1\n",
            "shown\n",
            "key\n",
            "mediator\n",
            "angiogenesis\n",
            "inflammation\n",
            "greater\n",
            "risk\n",
            "perioperative\n",
            "complications\n",
            "present\n",
            "25yearold\n",
            "woman\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "severe\n",
            "aortic\n",
            "insufficiency\n",
            "minimally\n",
            "invasive\n",
            "warm\n",
            "beating\n",
            "heart\n",
            "approach\n",
            "adopted\n",
            "mp\n",
            "plays\n",
            "important\n",
            "role\n",
            "etiology\n",
            "hypercoagulable\n",
            "state\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "well\n",
            "reduced\n",
            "red\n",
            "cell\n",
            "life\n",
            "span\n",
            "adhesive\n",
            "interactions\n",
            "red\n",
            "cells\n",
            "\n",
            "\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "stroketwentyfour\n",
            "percent\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "patients\n",
            "stroke\n",
            "age\n",
            "45\n",
            "years\n",
            "blood\n",
            "transfusions\n",
            "decrease\n",
            "stroke\n",
            "risk\n",
            "incidental\n",
            "results\n",
            "methods\n",
            "inform\n",
            "policy\n",
            "disclosure\n",
            "infant\n",
            "sickle\n",
            "cell\n",
            "disorder\n",
            "$\n",
            "carrier\n",
            "results\n",
            "mixedmethods\n",
            "study\n",
            "healthcare\n",
            "providers\n",
            "conducted\n",
            "ontario\n",
            "canada\n",
            "understand\n",
            "transfusiondependent\n",
            "patients\n",
            "thalassaemia\n",
            "major\n",
            "tm\n",
            "lesser\n",
            "extent\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "data\n",
            "available\n",
            "patients\n",
            "sicklebeta0thalassaemia\n",
            "iron\n",
            "deposition\n",
            "heart\n",
            "liver\n",
            "pancreas\n",
            "assessed\n",
            "length\n",
            "stay\n",
            "inhospital\n",
            "mortality\n",
            "hospitalizations\n",
            "patients\n",
            "without\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "undergoing\n",
            "highvolume\n",
            "surgical\n",
            "procedures\n",
            "used\n",
            "clinical\n",
            "classification\n",
            "software\n",
            "ccs\n",
            "codes\n",
            "individuals\n",
            "carriers\n",
            "prone\n",
            "immunization\n",
            "exposed\n",
            "normal\n",
            "antigens\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "patient\n",
            "alloimmunization\n",
            "major\n",
            "cause\n",
            "transfusion\n",
            "failure\n",
            "potential\n",
            "individuals\n",
            "partial\n",
            "c\n",
            "cell\n",
            "disease\n",
            "patientsdespite\n",
            "well\n",
            "known\n",
            "monogenic\n",
            "etiopathogenesis\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "characterized\n",
            "striking\n",
            "variability\n",
            "clinical\n",
            "presentation\n",
            "growing\n",
            "evidence\n",
            "genetic\n",
            "factors\n",
            "may\n",
            "common\n",
            "although\n",
            "hepatic\n",
            "iron\n",
            "overload\n",
            "recognized\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "cardiac\n",
            "iron\n",
            "deposition\n",
            "examined\n",
            "reports\n",
            "methods\n",
            "crosssectional\n",
            "study\n",
            "23\n",
            "patients\n",
            "standard\n",
            "care\n",
            "b\n",
            "thalassemia\n",
            "transfusions\n",
            "also\n",
            "indicated\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "patients\n",
            "hypersplenism\n",
            "recurrent\n",
            "vasoocclusive\n",
            "crises\n",
            "stroke\n",
            "prevention\n",
            "iron\n",
            "overload\n",
            "caused\n",
            "informants\n",
            "multiple\n",
            "measuresbackground\n",
            "behavioral\n",
            "emotional\n",
            "problems\n",
            "children\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "may\n",
            "related\n",
            "disease\n",
            "factors\n",
            "sociodemographic\n",
            "factors\n",
            "aim\n",
            "study\n",
            "investigate\n",
            "prevalence\n",
            "transfusion\n",
            "policytransfusion\n",
            "therapy\n",
            "may\n",
            "save\n",
            "lives\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "also\n",
            "associated\n",
            "high\n",
            "risk\n",
            "transmission\n",
            "infection\n",
            "aims\n",
            "study\n",
            "determine\n",
            "prevalence\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "diseasepulmonary\n",
            "hypertension\n",
            "serious\n",
            "complication\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "high\n",
            "morbidity\n",
            "mortality\n",
            "endothelin\n",
            "et1\n",
            "potent\n",
            "vasoconstrictor\n",
            "elevated\n",
            "scd\n",
            "acts\n",
            "large\n",
            "urban\n",
            "medicaid\n",
            "managed\n",
            "care\n",
            "organizationalthough\n",
            "patients\n",
            "sickle\n",
            "cell\n",
            "disease\n",
            "$\n",
            "hospitalized\n",
            "infrequently\n",
            "manage\n",
            "painful\n",
            "crises\n",
            "home\n",
            "small\n",
            "subpopulation\n",
            "frequently\n",
            "admitted\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "'''\n",
        "Basic LSTM\n",
        "'''\n",
        "import numpy as np\n",
        "import keras\n",
        "from nltk.tokenize import RegexpTokenizer\n",
        "from keras.models import Sequential, load_model\n",
        "from keras.layers import LSTM\n",
        "from keras.layers import Bidirectional\n",
        "from keras.layers import Dropout\n",
        "from keras.layers.core import Dense, Activation\n",
        "from keras.optimizers import RMSprop\n",
        "import matplotlib.pyplot as plt\n",
        "import pickle\n",
        "import heapq\n",
        "\n",
        "WindowSizeTot = WindowSize*2 +1\n",
        "ConvertedCorpus = np.array(ConvertedCorpus)# .reshape((1,WindowSize,len(ConvertedCorpus)))\n",
        "ConvertedLabels = np.array(ConvertedLabels)\n",
        "\n",
        "import tensorflow as tf\n",
        "\n",
        "ConvertedCorpus = tf.cast(ConvertedCorpus, tf.float32)\n",
        "ConvertedLabels = tf.cast(ConvertedLabels, tf.float32)\n",
        "\n",
        "# model = Sequential()\n",
        "# model.add(LSTM(128, input_shape=(WindowSize, 1)))\n",
        "# # model.add(Dense(len(TokenKey)))\n",
        "# model.add(Activation(activation='softmax'))\n",
        "\n",
        "# model.summary()\n",
        "\n",
        "# optimizer = RMSprop(lr=0.01)\n",
        "# model.compile(loss='categorical_crossentropy', optimizer=optimizer, metrics=['accuracy'])\n",
        "# history = model.fit(x=ConvertedCorpus, y=ConvertedLabels, validation_split=0.05, batch_size=128, epochs=2, shuffle=True).history\n",
        "\n",
        "model = keras.Sequential(\n",
        "    [\n",
        "        keras.Input(shape=(WindowSizeTot, 1)),\n",
        "        # keras.layers.LSTM(128),\n",
        "        Bidirectional(LSTM(128)),\n",
        "        keras.layers.Dense(410,activation=\"softmax\"),\n",
        "        # Dropout(.2, input_shape=(2,)),\n",
        "    ]\n",
        ")\n",
        "optimizer = keras.optimizers.RMSprop(learning_rate=0.01)\n",
        "model.compile(loss=\"categorical_crossentropy\", optimizer=optimizer)\n",
        "\n",
        "history = model.fit(x=ConvertedCorpus, y=ConvertedLabels, validation_split=0.1, batch_size=128, epochs=100, shuffle=True).history\n",
        "\n",
        "model.save('keras_next_word_model.h5')\n",
        "pickle.dump(history, open(\"history.p\", \"wb\"))\n",
        "model = load_model('keras_next_word_model.h5')\n",
        "history = pickle.load(open(\"history.p\", \"rb\"))\n",
        "\n",
        "plt.plot(history['loss'])\n",
        "plt.plot(history['val_loss'])\n",
        "plt.title('model accuracy')\n",
        "plt.ylabel('accuracy')\n",
        "plt.xlabel('epoch')\n",
        "plt.legend(['train', 'test'], loc='upper left')\n"
      ],
      "metadata": {
        "id": "26t5m4J3vNRF",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 520
        },
        "outputId": "86bdead7-ebc1-4637-94e7-f4b782a14568"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Epoch 1/100\n",
            "267/267 [==============================] - 13s 36ms/step - loss: 7.6523 - val_loss: 10.4274\n",
            "Epoch 2/100\n",
            "267/267 [==============================] - 6s 24ms/step - loss: 7.5804 - val_loss: 13.7649\n",
            "Epoch 3/100\n",
            "267/267 [==============================] - 8s 32ms/step - loss: 7.4679 - val_loss: 15.5085\n",
            "Epoch 4/100\n",
            "256/267 [===========================>..] - ETA: 0s - loss: 7.5210"
          ]
        },
        {
          "output_type": "error",
          "ename": "KeyboardInterrupt",
          "evalue": "ignored",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-25-e51b0d41c233>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     47\u001b[0m \u001b[0mmodel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloss\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"categorical_crossentropy\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moptimizer\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0moptimizer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     48\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 49\u001b[0;31m \u001b[0mhistory\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmodel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mConvertedCorpus\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mConvertedLabels\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalidation_split\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0.1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbatch_size\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m128\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mepochs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m100\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mshuffle\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhistory\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     50\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     51\u001b[0m \u001b[0mmodel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msave\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'keras_next_word_model.h5'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/utils/traceback_utils.py\u001b[0m in \u001b[0;36merror_handler\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m     62\u001b[0m     \u001b[0mfiltered_tb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m     \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 64\u001b[0;31m       \u001b[0;32mreturn\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     65\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# pylint: disable=broad-except\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     66\u001b[0m       \u001b[0mfiltered_tb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_process_traceback_frames\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__traceback__\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/engine/training.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_batch_size, validation_freq, max_queue_size, workers, use_multiprocessing)\u001b[0m\n\u001b[1;32m   1407\u001b[0m                 _r=1):\n\u001b[1;32m   1408\u001b[0m               \u001b[0mcallbacks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mon_train_batch_begin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstep\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1409\u001b[0;31m               \u001b[0mtmp_logs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtrain_function\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0miterator\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1410\u001b[0m               \u001b[0;32mif\u001b[0m \u001b[0mdata_handler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshould_sync\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1411\u001b[0m                 \u001b[0mcontext\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masync_wait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/callbacks.py\u001b[0m in \u001b[0;36mon_train_batch_end\u001b[0;34m(self, batch, logs)\u001b[0m\n\u001b[1;32m    436\u001b[0m     \"\"\"\n\u001b[1;32m    437\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_should_call_train_batch_hooks\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 438\u001b[0;31m       \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_batch_hook\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mModeKeys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTRAIN\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'end'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlogs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlogs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    439\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    440\u001b[0m   \u001b[0;32mdef\u001b[0m \u001b[0mon_test_batch_begin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlogs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/callbacks.py\u001b[0m in \u001b[0;36m_call_batch_hook\u001b[0;34m(self, mode, hook, batch, logs)\u001b[0m\n\u001b[1;32m    295\u001b[0m       \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_batch_begin_hook\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlogs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    296\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mhook\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'end'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 297\u001b[0;31m       \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_batch_end_hook\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlogs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    298\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    299\u001b[0m       raise ValueError(\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/callbacks.py\u001b[0m in \u001b[0;36m_call_batch_end_hook\u001b[0;34m(self, mode, batch, logs)\u001b[0m\n\u001b[1;32m    316\u001b[0m       \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_batch_times\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbatch_time\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    317\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 318\u001b[0;31m     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_batch_hook_helper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhook_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlogs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    319\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    320\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_batch_times\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_num_batches_for_timing_check\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/callbacks.py\u001b[0m in \u001b[0;36m_call_batch_hook_helper\u001b[0;34m(self, hook_name, batch, logs)\u001b[0m\n\u001b[1;32m    354\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mcallback\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcallbacks\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    355\u001b[0m       \u001b[0mhook\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcallback\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhook_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 356\u001b[0;31m       \u001b[0mhook\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbatch\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlogs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    357\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    358\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_timing\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/callbacks.py\u001b[0m in \u001b[0;36mon_train_batch_end\u001b[0;34m(self, batch, logs)\u001b[0m\n\u001b[1;32m   1032\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1033\u001b[0m   \u001b[0;32mdef\u001b[0m \u001b[0mon_train_batch_end\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlogs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1034\u001b[0;31m     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_batch_update_progbar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbatch\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlogs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1035\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1036\u001b[0m   \u001b[0;32mdef\u001b[0m \u001b[0mon_test_batch_end\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlogs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/callbacks.py\u001b[0m in \u001b[0;36m_batch_update_progbar\u001b[0;34m(self, batch, logs)\u001b[0m\n\u001b[1;32m   1105\u001b[0m       \u001b[0;31m# Only block async when verbose = 1.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1106\u001b[0m       \u001b[0mlogs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtf_utils\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msync_to_numpy_or_python_type\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlogs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1107\u001b[0;31m       \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprogbar\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseen\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlogs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfinalize\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1108\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1109\u001b[0m   \u001b[0;32mdef\u001b[0m \u001b[0m_finalize_progbar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlogs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcounter\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/utils/generic_utils.py\u001b[0m in \u001b[0;36mupdate\u001b[0;34m(self, current, values, finalize)\u001b[0m\n\u001b[1;32m    974\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    975\u001b[0m       \u001b[0mmessage\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0minfo\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 976\u001b[0;31m       \u001b[0mio_utils\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprint_msg\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmessage\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mline_break\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    977\u001b[0m       \u001b[0mmessage\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    978\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/keras/utils/io_utils.py\u001b[0m in \u001b[0;36mprint_msg\u001b[0;34m(message, line_break)\u001b[0m\n\u001b[1;32m     76\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     77\u001b[0m       \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrite\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmessage\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 78\u001b[0;31m     \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     79\u001b[0m   \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     80\u001b[0m     \u001b[0mlogging\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minfo\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmessage\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.7/dist-packages/ipykernel/iostream.py\u001b[0m in \u001b[0;36mflush\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    350\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpub_thread\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mschedule\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mevt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    351\u001b[0m                 \u001b[0;31m# and give a timeout to avoid\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 352\u001b[0;31m                 \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mevt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush_timeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    353\u001b[0m                     \u001b[0;31m# write directly to __stderr__ instead of warning because\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    354\u001b[0m                     \u001b[0;31m# if this is happening sys.stderr may be the problem.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/lib/python3.7/threading.py\u001b[0m in \u001b[0;36mwait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    550\u001b[0m             \u001b[0msignaled\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_flag\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    551\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0msignaled\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 552\u001b[0;31m                 \u001b[0msignaled\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_cond\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    553\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0msignaled\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    554\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/lib/python3.7/threading.py\u001b[0m in \u001b[0;36mwait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    298\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    299\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mtimeout\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 300\u001b[0;31m                     \u001b[0mgotit\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mwaiter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0macquire\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    301\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    302\u001b[0m                     \u001b[0mgotit\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mwaiter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0macquire\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "'''\n",
        "Data visualization\n",
        "'''\n",
        "import numpy as np\n",
        "\n",
        "# test = {}\n",
        "# test['test1'] = 0\n",
        "# print(test)\n",
        "\n",
        "# print(len(TokenKey))\n",
        "# input_shape=(WindowSize, len(TokenKey))\n",
        "# print(input_shape)\n",
        "# y = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]).reshape((5,2))\n",
        "# print(y)\n",
        "\n",
        "# X = np.zeros((len(WindowedCorpus), WindowSize, len(TokenKey)), dtype=bool)\n",
        "\n",
        "# print(np.zeros((5,4,11)))\n",
        "\n",
        "Test = [1, 2, [3, 4]]\n",
        "print(Test)\n",
        "print(Test[2][1])\n",
        "print(history.keys())"
      ],
      "metadata": {
        "id": "DLM2Jf04RnSX",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "00f08cbe-8145-4b6c-9647-51788c486604"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[1, 2, [3, 4]]\n",
            "4\n",
            "dict_keys(['loss', 'val_loss'])\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "'''\n",
        "Models\n",
        "'''\n",
        "\n",
        "# LSTM\n",
        "# BiLSTM\n",
        "\n"
      ],
      "metadata": {
        "id": "yiQ9o4wx6C6S"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "'''\n",
        "Train WSD\n",
        "'''\n",
        "# Use seed to make reproducable\n",
        "# What approach do I use?\n",
        "# Test various algos"
      ],
      "metadata": {
        "id": "MQTmxo8SfxAi"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "'''\n",
        "Test WSD\n",
        "'''\n",
        "# Use k-fold validation"
      ],
      "metadata": {
        "id": "OsbRaDrHf3th"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "'''\n",
        "Implement WSD\n",
        "'''\n",
        "# Search through the data in the cancer corpus for ambiguous words\n",
        "# Use the WSD to get the CUI code for that ambiguous word\n",
        "# Replace that ambiguous instance with the CUI code to pass the document on to the document classifier"
      ],
      "metadata": {
        "id": "gzlrKg4Sf6fW"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# https://gweissman.github.io/post/using-metamap-with-python-to-access-the-umls-metathesaurus-a-quick-start-guide/\n",
        "# https://documentation.uts.nlm.nih.gov/rest/search/\n",
        "\n"
      ],
      "metadata": {
        "id": "CO_A4WjM2Hbg"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}